{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# S2 explorations\n",
    "\n",
    "## General notes on data\n",
    "- DOIs are missing. \n",
    "  -  if we want to find SAGE papers which are related to Coronavirus, we will need to get the DOIs from the S2 API and then get all the data on the DOIs from CrossRef.\n",
    "- Text data is quite consistent, but there are some missing abstracts. \n",
    "  - Haven't checked full-text data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Download the data\n",
    "#!curl https://ai2-semanticscholar-cord-19.s3-us-west-2.amazonaws.com/2020-03-13/comm_use_subset.tar.gz --output data/comm_use_subset.tar.gz\n",
    "#!curl https://ai2-semanticscholar-cord-19.s3-us-west-2.amazonaws.com/2020-03-13/noncomm_use_subset.tar.gz --output data/noncomm_use_subset.tar.gz \n",
    "#!curl https://ai2-semanticscholar-cord-19.s3-us-west-2.amazonaws.com/2020-03-13/pmc_custom_license.tar.gz --output data/pmc_custom_license.tar.gz\n",
    "#!curl https://ai2-semanticscholar-cord-19.s3-us-west-2.amazonaws.com/2020-03-13/biorxiv_medrxiv.tar.gz --output data/biorxiv_medrxiv.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import tarfile\n",
    "import json\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "pull the data from tar files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wall time: 23 s\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(13202,\n",
       " ['biorxiv_medrxiv.tar.gz',\n",
       "  'comm_use_subset.tar.gz',\n",
       "  'noncomm_use_subset.tar.gz',\n",
       "  'pmc_custom_license.tar.gz'])"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "data = []\n",
    "\n",
    "sets = [x for x in os.listdir('data') if '.tar.gz' in x]\n",
    "\n",
    "for set_ in sets:\n",
    "    with tarfile.open(set_, \"r:gz\") as tar:\n",
    "    #     print(tar.getmembers())\n",
    "        for i,member in enumerate(tar.getmembers()):\n",
    "            f=tar.extractfile(member)\n",
    "            if f is not None:\n",
    "                data.append(json.load(f)) #.read())\n",
    "len(data), sets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### quick consistency check\n",
    "Just looking for text data at the moment. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paper_id': 13202,\n",
       " 'metadata': 13202,\n",
       " 'abstract': 13202,\n",
       " 'body_text': 13202,\n",
       " 'bib_entries': 13202,\n",
       " 'ref_entries': 13202,\n",
       " 'back_matter': 13202}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# how many have titles, how many have abstracts\n",
    "counts = {}\n",
    "for item in data:\n",
    "    for key in item:\n",
    "        if key not in counts:\n",
    "            counts[key]=1\n",
    "        else:\n",
    "            counts[key]+=1\n",
    "counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13202"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# unique papers\n",
    "len(list(set([x['paper_id'] for x in data])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12099"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# titles?\n",
    "len(list(set([x['metadata']['title'] for x in data])))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some titles are not unique. Causes:\n",
    "- missing titles\n",
    "- article types as titles (like 'opinion piece')\n",
    "- duplication\n",
    "Duplication seems like a relatively minor feature of the data. None of the other duplicated titles seem like they would create a significant problem in the text-data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('', 943),\n",
       " ('CLINICAL EXPERIMENTAL VACCINE RESEARCH', 14),\n",
       " ('Access to', 8),\n",
       " ('Morbidity and Mortality Weekly Report', 8),\n",
       " ('Veterinary Science pISSN 1229-845X, eISSN 1976-555X', 7),\n",
       " ('Infection & Chemotherapy', 7),\n",
       " ('PEER REVIEW HISTORY', 6),\n",
       " ('Supporting Information', 5),\n",
       " ('Epidemiology and Infection', 5),\n",
       " ('INQUIRY: The Journal of Health Care Organization, Provision, and Financing',\n",
       "  5),\n",
       " ('Original Article', 4),\n",
       " ('Comment', 4),\n",
       " ('Supplementary Figures', 4),\n",
       " ('Brief Communication', 4),\n",
       " ('Advance Access Publication', 4),\n",
       " ('Ar ticle', 4),\n",
       " ('EMERGENCE OF SARS', 4),\n",
       " ('DISPATCHES', 4),\n",
       " ('Rapid communication', 3),\n",
       " ('Supplementary Information', 3),\n",
       " ('Surveillance and outbreak report', 3),\n",
       " ('Opinion piece', 3),\n",
       " ('LETTERS', 3),\n",
       " ('Search past issues of EID at www.cdc.gov/eid', 3),\n",
       " ('World Health Organization Methodology to Prioritize Emerging Infectious Diseases in Need of Research and Development',\n",
       "  3),\n",
       " ('Bayesian mixture analysis for metagenomic community profiling', 2),\n",
       " ('MERS-CoV recombination: implications about the reservoir and potential for adaptation',\n",
       "  2),\n",
       " ('Quantifying the Risk and Cost of Active Monitoring for Infectious Diseases',\n",
       "  2),\n",
       " ('MERS-CoV spillover at the camel-human interface', 2),\n",
       " ('TWIRLS, an automated topic-wise inference method based on massive literature, suggests a possible mechanism via ACE2 for the pathological changes in the human host after coronavirus infection',\n",
       "  2)]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# which ones aren't unique\n",
    "from collections import Counter\n",
    "Counter([x['metadata']['title'] for x in data]).most_common(30)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some abstracts are missing also. It's a small number, so we can probably just ignore these...\n",
    "\n",
    "If we want to include them, perhaps just take the first para of full-text instead?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13202, 13202, 13202)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# how many missing abstracts?\n",
    "titles = []\n",
    "abstracts = []\n",
    "tiabss = []\n",
    "i=0\n",
    "for x in data:\n",
    "    title = x['metadata']['title']\n",
    "    titles.append(title)\n",
    "    abstract = x['abstract']\n",
    "    abs_text = ''\n",
    "    if type(abstract) == list and len(abstract)>0:\n",
    "        for part in abstract:\n",
    "            if 'text' in part and len(abs_text)==0:\n",
    "                abs_text = part['text']\n",
    "            else:\n",
    "                abs_text = abs_text+'. '+part['text']\n",
    "    abstracts.append(abs_text)\n",
    "    tiabss.append(title+'. '+abs_text)\n",
    "    \n",
    "\n",
    "len(titles), len(abstracts), len(tiabss)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The cell below is for exploring all the articles with missing title/abstract. THere are definitely some issues in the text data for some percentage of articles. \n",
    "\n",
    "For simplicity drop everything without a typical title + abstract."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# m =0\n",
    "# for i,x in enumerate(tiabss): \n",
    "#     if len(x)<10:\n",
    "#         print(data[i])\n",
    "#         print()\n",
    "#         print('------------')\n",
    "#         print()\n",
    "#         m+=1\n",
    "#         if m>5:\n",
    "#             break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# data[i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x1a05ed134c8>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXcAAAD4CAYAAAAXUaZHAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAXVUlEQVR4nO3dfYxc1XnH8e+vOLyETWwDYWvZVpcKK02EGwdG4JSq2sVJZUgV8wdUiazEIFfbP0hKGqJgWqlRpFZ1VBECNEJdxVFM5bIQErSWIS+WYVohFSdeQlgTh3qhDlnsektsNt1A2pA8/WPOOsN61nt3XnZ3zvw+0ujee+65d84zM/vcs2fu3KuIwMzM8vJbC90AMzNrPid3M7MMObmbmWXIyd3MLENO7mZmGVqy0A0AuOiii6Knp6eubX/+859z/vnnN7dBi5xj7gyOuTM0EvPw8PArEfGOWusWRXLv6enhwIEDdW1bLpfp7e1tboMWOcfcGRxzZ2gkZkk/nmmdh2XMzDLk5G5mliEndzOzDDm5m5llyMndzCxDTu5mZhlycjczy5CTu5lZhgold0l/Kek5SQclPSDpXEmXSNov6bCkByWdneqek5ZH0/qeVgZgZmanmzW5S1oJ/AVQiojLgLOADwOfB+6KiDXASWBr2mQrcDIiLgXuSvVaZuTlCXq2PUrPtkdb+TRmZm2l6LDMEuA8SUuAtwLHgGuAh9P6ncD1aX5TWiat3yBJzWmumZkVoSK32ZN0K/B3wOvAd4BbgadS7xxJq4FvRsRlkg4CGyNiLK17AbgqIl6Zts9+oB+gu7v7isHBwboCGD8xwfHXK/NrVy6tax/tZnJykq6uroVuxrxyzJ3BMc9NX1/fcESUaq2b9cJhkpZT6Y1fArwKfA24tkbVqaNErV76aUeQiBgABgBKpVLUe+Gce3cNcedIJYwjm+vbR7vxxZU6g2PuDK2KuciwzPuB/4yI/46IXwLfAP4AWJaGaQBWAUfT/BiwGiCtXwqcaGqrzczsjIok95eA9ZLemsbONwA/BJ4Abkh1tgBDaX53WiatfzyKjP2YmVnTzJrcI2I/lS9GnwZG0jYDwO3ApySNAhcCO9ImO4ALU/mngG0taLeZmZ1BoZt1RMRngc9OK34RuLJG3V8ANzbeNDMzq5d/oWpmliEndzOzDDm5m5llyMndzCxDTu5mZhlycjczy5CTu5lZhpzczcwy5ORuZpYhJ3czsww5uZuZZcjJ3cwsQ07uZmYZcnI3M8uQk7uZWYac3M3MMjRrcpf0TknPVD1+JumTki6QtFfS4TRdnupL0j2SRiU9K+ny1odhZmbVitxm7/mIWBcR64ArgNeAR6jcPm9fRKwB9vGb2+ldC6xJj37gvlY03MzMZjbXYZkNwAsR8WNgE7Azle8Erk/zm4D7o+IpYJmkFU1prZmZFaKIKF5Z+grwdET8o6RXI2JZ1bqTEbFc0h5ge0Q8mcr3AbdHxIFp++qn0rOnu7v7isHBwboCGD8xwfHXK/NrVy6tax/tZnJykq6uroVuxrxyzJ3BMc9NX1/fcESUaq0rdINsAElnAx8C7pitao2y044gETEADACUSqXo7e0t2pQ3uXfXEHeOVMI4srm+fbSbcrlMva9Xu3LMncExN89chmWupdJrP56Wj08Nt6TpeCofA1ZXbbcKONpoQ83MrLi5JPePAA9ULe8GtqT5LcBQVfnH0lkz64GJiDjWcEvNzKywQsMykt4KfAD486ri7cBDkrYCLwE3pvLHgOuAUSpn1tzctNaamVkhhZJ7RLwGXDit7KdUzp6ZXjeAW5rSOjMzq4t/oWpmliEndzOzDDm5m5llyMndzCxDTu5mZhlycjczy5CTu5lZhpzczcwy5ORuZpYhJ3czsww5uZuZZcjJ3cwsQ07uZmYZcnI3M8uQk7uZWYac3M3MMuTkbmaWoULJXdIySQ9L+pGkQ5LeJ+kCSXslHU7T5amuJN0jaVTSs5Iub20IZmY2XdGe+93AtyLi94D3AIeAbcC+iFgD7EvLANcCa9KjH7ivqS02M7NZzZrcJb0d+CNgB0BE/F9EvApsAnamajuB69P8JuD+qHgKWCZpRdNbbmZmM1LlftZnqCCtAwaAH1LptQ8DtwIvR8SyqnonI2K5pD3A9oh4MpXvA26PiAPT9ttPpWdPd3f3FYODg3UFMH5iguOvV+bXrlxa1z7azeTkJF1dXQvdjHnlmDuDY56bvr6+4Ygo1Vq3pMD2S4DLgU9ExH5Jd/ObIZhaVKPstCNIRAxQOWhQKpWit7e3QFNOd++uIe4cqYRxZHN9+2g35XKZel+vduWYO4Njbp4iY+5jwFhE7E/LD1NJ9senhlvSdLyq/uqq7VcBR5vTXDMzK2LW5B4R/wX8RNI7U9EGKkM0u4EtqWwLMJTmdwMfS2fNrAcmIuJYc5ttZmZnUmRYBuATwC5JZwMvAjdTOTA8JGkr8BJwY6r7GHAdMAq8luqamdk8KpTcI+IZoNag/YYadQO4pcF2mZlZA/wLVTOzDDm5m5llyMndzCxDTu5mZhlycjczy5CTu5lZhpzczcwy5ORuZpYhJ3czsww5uZuZZcjJ3cwsQ07uZmYZcnI3M8uQk7uZWYac3M3MMuTkbmaWoULJXdIRSSOSnpF0IJVdIGmvpMNpujyVS9I9kkYlPSvp8lYGYGZmp5tLz70vItZFxNQdmbYB+yJiDbAvLQNcC6xJj37gvmY11szMimlkWGYTsDPN7wSuryq/PyqeApZJWtHA85iZ2RwVTe4BfEfSsKT+VNYdEccA0vTiVL4S+EnVtmOpzMzM5kmhG2QDV0fEUUkXA3sl/egMdVWjLE6rVDlI9AN0d3dTLpcLNuXNus+D29a+AVD3PtrN5ORkx8Q6xTF3BsfcPIWSe0QcTdNxSY8AVwLHJa2IiGNp2GU8VR8DVldtvgo4WmOfA8AAQKlUit7e3roCuHfXEHeOVMI4srm+fbSbcrlMva9Xu3LMncExN8+swzKSzpf0tql54I+Bg8BuYEuqtgUYSvO7gY+ls2bWAxNTwzdmZjY/ivTcu4FHJE3V/5eI+Jak7wEPSdoKvATcmOo/BlwHjAKvATc3vdVmZnZGsyb3iHgReE+N8p8CG2qUB3BLU1pnZmZ18S9Uzcwy5ORuZpahoqdCtp2ebY+emj+y/YML2BIzs/nnnruZWYac3M3MMuTkbmaWISd3M7MMObmbmWXIyd3MLENO7mZmGXJyNzPLkJO7mVmGnNzNzDLk5G5mliEndzOzDDm5m5llyMndzCxDhZO7pLMkfV/SnrR8iaT9kg5LelDS2an8nLQ8mtb3tKbpZmY2k7n03G8FDlUtfx64KyLWACeBral8K3AyIi4F7kr1zMxsHhVK7pJWAR8EvpyWBVwDPJyq7ASuT/Ob0jJp/YZU38zM5okq97OepZL0MPD3wNuATwM3AU+l3jmSVgPfjIjLJB0ENkbEWFr3AnBVRLwybZ/9QD9Ad3f3FYODg3UFMH5iguOvV+bXrlx6qnzk5YlT89XlOZicnKSrq2uhmzGvHHNncMxz09fXNxwRpVrrZr3NnqQ/AcYjYlhS71RxjapRYN1vCiIGgAGAUqkUvb2906sUcu+uIe4cqYRxZPNv9nFT9W32qspzUC6Xqff1aleOuTM45uYpcg/Vq4EPSboOOBd4O/BFYJmkJRHxBrAKOJrqjwGrgTFJS4ClwImmt9zMzGY0a3KPiDuAOwBSz/3TEbFZ0teAG4BBYAswlDbZnZb/Pa1/PIqM/TRB9U2xzcw6WSPnud8OfErSKHAhsCOV7wAuTOWfArY11kQzM5urIsMyp0REGSin+ReBK2vU+QVwYxPaZmZmdfIvVM3MMuTkbmaWISd3M7MMObmbmWXIyd3MLENO7mZmGXJyNzPLkJO7mVmGnNzNzDLk5G5mliEndzOzDDm5m5llaE4XDjNbbKov83xk+wcXsCVmi4t77mZmGXJyNzPLkJO7mVmGnNzNzDI0a3KXdK6k70r6gaTnJH0ulV8iab+kw5IelHR2Kj8nLY+m9T2tDcHMzKYr0nP/X+CaiHgPsA7YKGk98HngrohYA5wEtqb6W4GTEXEpcFeqZ2Zm82jW5B4Vk2nxLekRwDXAw6l8J3B9mt+UlknrN0hS01psNoOebY+eeph1OkXE7JWks4Bh4FLgS8A/AE+l3jmSVgPfjIjLJB0ENkbEWFr3AnBVRLwybZ/9QD9Ad3f3FYODg3UFMH5iguOvn7nO2pVL69r3YjU5OUlXV9dCN2NezRTzyMsTNevn8J77fe4MjcTc19c3HBGlWusK/YgpIn4FrJO0DHgEeFetamlaq5d+2hEkIgaAAYBSqRS9vb1FmnKae3cNcefImcM4srm+fS9W5XKZel+vdjVTzDfN0EvP4T33+9wZWhXznM6WiYhXgTKwHlgmaSqrrgKOpvkxYDVAWr8UONGMxpqZWTFFzpZ5R+qxI+k84P3AIeAJ4IZUbQswlOZ3p2XS+sejyNiPmZk1TZFhmRXAzjTu/lvAQxGxR9IPgUFJfwt8H9iR6u8A/lnSKJUe+4db0O458fVHzKzTzJrcI+JZ4L01yl8ErqxR/gvgxqa0zszM6uJfqJqZZcjJ3cwsQ76eu7Ud/0jJbHZO7pYlf4lunc7J3axOPoDYYuYxdzOzDLnnbjYHHu+3duHkbm1h5OWJGa8jY2anc3K3pvD4s9ni4uRu1gQ+uNli4y9Uzcwy5ORuZpYhD8vYorIYhjcWQxvMGuWeu5lZhtxzt+y5J26dyD13M7MMzdpzl7QauB/4beDXwEBE3C3pAuBBoAc4AvxpRJyUJOBu4DrgNeCmiHi6Nc23nFX3uG9b29r9m+WmyLDMG8BtEfG0pLcBw5L2AjcB+yJiu6RtwDbgduBaYE16XAXcl6ZmHcdDQrZQitxm7xhwLM3/j6RDwEpgE9Cbqu0EylSS+ybg/nRT7KckLZO0Iu3HLHv+j8AWgzmNuUvqoXI/1f1A91TCTtOLU7WVwE+qNhtLZWZmNk9U6WAXqCh1Af8K/F1EfEPSqxGxrGr9yYhYLulR4O8j4slUvg/4TEQMT9tfP9AP0N3dfcXg4GBdAYyfmOD468Xrr125tK7nWUwmJyfp6upa6Ga8ycjLE6fmG3mNq/dTrfs85vQ+z6S6bTM910z1i25TdF+zWYzvc6s55rnp6+sbjohSrXWFToWU9Bbg68CuiPhGKj4+NdwiaQUwnsrHgNVVm68Cjk7fZ0QMAAMApVIpent7izTlNPfuGuLOkeJndB7ZXN/zLCblcpl6X69mevPwQ9V7MPLzmvWLjDnPdOXH29a+Maf3eSbV73+Rq0xO/7w0cmXKuX72Fsv7PJ8cc/MUOVtGwA7gUER8oWrVbmALsD1Nh6rKPy5pkMoXqRMeb8+Hx5PN2kORrtDVwEeBEUnPpLK/opLUH5K0FXgJuDGte4zKaZCjVE6FvLmpLbbs+IAxNz4Dx4oocrbMk4BmWL2hRv0AbmmwXWYtMdcDiQ881q78C1Uzsww5uZuZZcjJ3cwsQ74qpNki5ZuCWyOc3G3edPpZHp0ev80vJ3ezBeakb63gMXczsww5uZuZZcjDMrYgOv3HQZ0ev7Wek7vZIjLXu095vN5m4mEZM7MMObmbmWWo44Zlpo91+l/Z2Xl82Kz9dFxyt2Kc0NuPx9+tmodlzMwy5ORuZpYhJ3czswzNmtwlfUXSuKSDVWUXSNor6XCaLk/lknSPpFFJz0q6vJWNNzOz2or03L8KbJxWtg3YFxFrgH1pGeBaYE169AP3NaeZ1go92x499TCzvBS5h+q/SeqZVrwJ6E3zO4EycHsqvz/dR/UpScskrYiIY81qsLWGE3xeWnHmjM/GaS+q5OFZKlWS+56IuCwtvxoRy6rWn4yI5ZL2ANvTTbWRtA+4PSIO1NhnP5XePd3d3VcMDg7WFcD4iQmOv17XpgCsXbm0/o0XyOTkJF1dXQ3vZ+TliSa0Zn50n0dD73M7albMM33Gq9//In8Hc61fj2Z9tttJIzH39fUNR0Sp1rpmn+euGmU1jx4RMQAMAJRKpejt7a3rCe/dNcSdI/WHcWRzfc873Xz2asrlMvW+XtXa6S4/t619o6H3uR01K+bqz/ib/0NbUrPOTKo/L836u5muWZ/tdtKqmOs9W+a4pBUAaTqeyseA1VX1VgFH62+emZnVo95uwW5gC7A9TYeqyj8uaRC4CpjweLvZwmrk+xR/F9O+Zk3ukh6g8uXpRZLGgM9SSeoPSdoKvATcmKo/BlwHjAKvATe3oM02R/4Dtdk0+hnxl62LT5GzZT4yw6oNNeoGcEujjWp38/lB9x+VmdXSWd9QdRD31s06W8cnd/d8zebOnYfFr+OTe7V2T/T+gzOzKU7uDZjPZFr9XF/deP68Pa+ZtSdfFdLMLEPuubdYI0M9M/1nMPLyRFv9utQ6S7sPb+bCPXczswy5575A3Lsxs1Zycp8jn5FiVh93aOaXk/sMWn097EbqmJnNxsl9EXBCN7Nmc3I3s3k3U4fGv+FoHif3AtyzNqvPYvvb6aRxfyd3M1s0qn/DMZ/fdeXIyd3MFqW5JuXce+Jz5eRuZtmZ64FhpvpFDhhneq6Ztp+Pa0U5uZtZFpqV0BvZ55m2n+//LFqS3CVtBO4GzgK+HBHbW/E8ZmbtYr7H/pt+bRlJZwFfAq4F3g18RNK7m/08ZmY2s1ZcOOxKYDQiXoyI/wMGgU0teB4zM5uBKve0buIOpRuAjRHxZ2n5o8BVEfHxafX6gf60+E7g+Tqf8iLglTq3bVeOuTM45s7QSMy/ExHvqLWiFWPuqlF22hEkIgaAgYafTDoQEaVG99NOHHNncMydoVUxt2JYZgxYXbW8CjjagucxM7MZtCK5fw9YI+kSSWcDHwZ2t+B5zMxsBk0flomINyR9HPg2lVMhvxIRzzX7eao0PLTThhxzZ3DMnaElMTf9C1UzM1t4voeqmVmGnNzNzDLUtsld0kZJz0salbRtodvTLJK+Imlc0sGqsgsk7ZV0OE2Xp3JJuie9Bs9KunzhWl4/SaslPSHpkKTnJN2ayrONW9K5kr4r6Qcp5s+l8ksk7U8xP5hOSkDSOWl5NK3vWcj2N0LSWZK+L2lPWu6EmI9IGpH0jKQDqayln++2TO6ZX+Lgq8DGaWXbgH0RsQbYl5ahEv+a9OgH7punNjbbG8BtEfEuYD1wS3o/c477f4FrIuI9wDpgo6T1wOeBu1LMJ4Gtqf5W4GREXArcleq1q1uBQ1XLnRAzQF9ErKs6p721n++IaLsH8D7g21XLdwB3LHS7mhhfD3Cwavl5YEWaXwE8n+b/CfhIrXrt/ACGgA90StzAW4Gngauo/FJxSSo/9TmncvbZ+9L8klRPC932OmJdlRLZNcAeKj96zDrm1P4jwEXTylr6+W7LnjuwEvhJ1fJYKstVd0QcA0jTi1N5dq9D+tf7vcB+Mo87DU88A4wDe4EXgFcj4o1UpTquUzGn9RPAhfPb4qb4IvAZ4Ndp+ULyjxkqv9L/jqThdOkVaPHnu12v517oEgcdIKvXQVIX8HXgkxHxM6lWeJWqNcraLu6I+BWwTtIy4BHgXbWqpWnbxyzpT4DxiBiW1DtVXKNqNjFXuToijkq6GNgr6UdnqNuUuNu1595plzg4LmkFQJqOp/JsXgdJb6GS2HdFxDdScfZxA0TEq0CZyvcNyyRNdbqq4zoVc1q/FDgxvy1t2NXAhyQdoXK12Guo9ORzjhmAiDiapuNUDuRX0uLPd7sm9067xMFuYEua30JlTHqq/GPp2/X1wMTUv3ntRJUu+g7gUER8oWpVtnFLekfqsSPpPOD9VL5kfAK4IVWbHvPUa3ED8HikAdl2ERF3RMSqiOih8jf7eERsJuOYASSdL+ltU/PAHwMHafXne6G/aGjgC4rrgP+gMk751wvdnibG9QBwDPgllSP4VirjjPuAw2l6QaorKmcNvQCMAKWFbn+dMf8hlX87nwWeSY/rco4b+H3g+ynmg8DfpPLfBb4LjAJfA85J5eem5dG0/ncXOoYG4+8F9nRCzCm+H6THc1P5qtWfb19+wMwsQ+06LGNmZmfg5G5mliEndzOzDDm5m5llyMndzCxDTu5mZhlycjczy9D/A8p7ODT/dg+kAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# what is the distribution of title+abstract lengths like?\n",
    "pd.Series([len(x.split()) for x in tiabss if len(x.split())<500]).hist(bins = 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1.0, 966.9899999999998)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# what is 99th percentile? and 1st percentile\n",
    "# perhaps drop everything outside of these ranges so that we don't have too many outliers?\n",
    "pd.Series([len(x.split()) for x in tiabss]).quantile(.04),pd.Series([len(x.split()) for x in tiabss]).quantile(.99)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10222"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len([x for x in tiabss if len(x.split())>50 and len(x.split())<500])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "No data on publication venue, DOI etc \n",
    "- can't use this to find SAGE articles.\n",
    "- might be able to make a classifier work with it though, but you need another dataset for that."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wall time: 1min 36s\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(12640, 1581)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "# There don't seem to be proper IDs in the data (DOIs/PMIDs etc)\n",
    "# let's get those from S2's API\n",
    "# see docs: https://api.semanticscholar.org/\n",
    "import requests\n",
    "import time\n",
    "\n",
    "retry_failures = False\n",
    "\n",
    "api_data = []\n",
    "failures = []\n",
    "api_dict = dict()\n",
    "filepath = 'data/s2_api_data.json'\n",
    "failpath = 'data/s2_api_data_failures.json'\n",
    "\n",
    "# read api data from file from past iterations\n",
    "if os.path.exists(filepath) and os.path.exists(failpath):\n",
    "    with open(filepath, 'r') as f:\n",
    "        api_data = json.load(f)\n",
    "        api_dict = {x['paperId']: x for x in api_data }\n",
    "    with open(failpath, 'r') as f:\n",
    "        failures = json.load(f)\n",
    "\n",
    "\n",
    "if retry_failures == True or len(api_data)==0:        \n",
    "    failures = set(failures)\n",
    "    base_url = 'https://api.semanticscholar.org/v1/paper/'\n",
    "    paper_ids = [x['paper_id'] for x in data]\n",
    "    i=0\n",
    "    for paper_id in [x for x in paper_ids if x not in api_dict]:\n",
    "        url = base_url+paper_id\n",
    "        resp = requests.get(url)\n",
    "        if resp.elapsed.total_seconds()>2:\n",
    "            time.sleep(2)\n",
    "        if resp.status_code==200:\n",
    "            js = resp.json()\n",
    "            api_data.append(js)\n",
    "        else:\n",
    "            if paper_id not in failures:\n",
    "                failures.add(paper_id)\n",
    "            \n",
    "with open(filepath,'w') as f:\n",
    "    json.dump(api_data,f)\n",
    "with open(failpath,'w') as f:\n",
    "    json.dump(list(failures),f)\n",
    "    \n",
    "len(api_data), len(failures)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exploration\n",
    "- Would be good to do some clustering on the text data to get a better look at it.\n",
    "- also look at publication dates.  What rate is content coming out at now?\n",
    "- which ones are SAGE papers - we should drop the paywall on them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paper_id': 'f056da9c64fbf00a4645ae326e8a4339d015d155',\n",
       " 'metadata': {'title': 'SIANN: Strain Identification by Alignment to Near Neighbors',\n",
       "  'authors': [{'first': 'Samuel',\n",
       "    'middle': ['S'],\n",
       "    'last': 'Minot',\n",
       "    'suffix': '',\n",
       "    'affiliation': {},\n",
       "    'email': ''},\n",
       "   {'first': 'Stephen',\n",
       "    'middle': ['D'],\n",
       "    'last': 'Turner',\n",
       "    'suffix': '',\n",
       "    'affiliation': {},\n",
       "    'email': ''},\n",
       "   {'first': 'Krista',\n",
       "    'middle': ['L'],\n",
       "    'last': 'Ternus',\n",
       "    'suffix': '',\n",
       "    'affiliation': {},\n",
       "    'email': ''},\n",
       "   {'first': 'Dana',\n",
       "    'middle': ['R'],\n",
       "    'last': 'Kadavy',\n",
       "    'suffix': '',\n",
       "    'affiliation': {},\n",
       "    'email': ''}]},\n",
       " 'abstract': [{'text': 'Next-generation sequencing is increasingly being used to study samples composed of mixtures of organisms, such as in clinical applications where the presence of a pathogen at very low abundance may be highly important. We present an analytical method (SIANN: Strain Identification by Alignment to Near Neighbors) specifically designed to rapidly detect a set of target organisms in mixed samples that achieves a high degree of species-and strain-specificity by aligning short sequence reads to the genomes of near neighbor organisms, as well as that of the target. Empirical benchmarking alongside the current state-of-the-art methods shows an extremely high Positive Predictive Value, even at very low abundances of the target organism in a mixed sample. SIANN is available as an Illumina BaseSpace app, as well as through Signature Science, LLC. SIANN results are presented in a streamlined report designed to be comprehensible to the non-specialist user, providing a powerful tool for rapid species detection in a mixed sample. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Abstract'}],\n",
       " 'body_text': [{'text': 'There are many different methods that characterize the mixture of organisms present within a metagenomic dataset. Such datasets are generated when a complex environmental sample is processed by a \"next-generation\" high-throughput genome sequencing protocol, and they consist of large numbers of short nucleotide sequences. Each sequence represents a small fragment of a randomly selected genome from the very large collection of genomes present in the source sample. Those sequences indicate the presence of one organism or another according to their similarity to a set of known reference genomes. While a given sequence may be unique to one species, it also may be found in diverse organisms across the tree of life. Therefore, one analytical challenge (among many) is to take that collection of sequences (likely numbering in the millions) and accurately determine what species are present in the sample. Here we describe a novel method (SIANN: Strain Identification by Alignment to Near Neighbors) that is specifically designed to rapidly detect a set of targeted organisms from a metagenomic dataset by aligning reads to genomic regions that are unique at the strain or species level.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Introduction'},\n",
       "  {'text': \"The analytical question motivating a particular piece of metagenomic bioinformatic analysis may vary widely by user and sample type (Segata, et al., 2013) . For example, the function of the human gut microbiome may depend on the relative abundance of hundreds of species of bacteria and the types of metabolic genes they contain (Wu, et al., 2011; Schloissnig, et al., 2013) . In contrast, the clinical treatment of a patient may depend on whether or not a particular virus, or a consortium of co-infecting pathogens, is/are detected in their blood. It is this second class of presence/absence questions that SIANN is designed to address. SIANN is appropriate for situations in which a user wants to know whether a particular organism or set of organisms is present in a sample, but isn't interested in the functions encoded in their genomes, the relative abundance of each organism, or any other more in-depth analysis.\",\n",
       "   'cite_spans': [{'start': 132,\n",
       "     'end': 154,\n",
       "     'text': '(Segata, et al., 2013)',\n",
       "     'ref_id': None},\n",
       "    {'start': 329, 'end': 347, 'text': '(Wu, et al., 2011;', 'ref_id': None},\n",
       "    {'start': 348,\n",
       "     'end': 374,\n",
       "     'text': 'Schloissnig, et al., 2013)',\n",
       "     'ref_id': 'BIBREF10'}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Introduction'},\n",
       "  {'text': 'Metagenomic classification methods are based on a wide variety of theoretical underpinnings. The basic varieties include alignment of reads to various nucleotide databases or exact matching to nucleotide or protein signature sequences (or kmers). A representative set of recent methods are described in Table 1 (also see Bazinet & Cummings 2012) .',\n",
       "   'cite_spans': [{'start': 321,\n",
       "     'end': 345,\n",
       "     'text': 'Bazinet & Cummings 2012)',\n",
       "     'ref_id': 'BIBREF1'}],\n",
       "   'ref_spans': [{'start': 303,\n",
       "     'end': 310,\n",
       "     'text': 'Table 1',\n",
       "     'ref_id': None}],\n",
       "   'section': 'Approach'},\n",
       "  {'text': 'Alignment to large nucleotide database Huson, et al., 2011 PhymmBL Alignment to large nucleotide database with interpolated Markov models Brady & Salzberg, 2011 Metaphyler Alignment to clade-specific marker genes Liu, et al., Overall, these methods are designed to either classify individual reads to, and/or predict the total abundance of, clades (e.g. genus or species) across the entire tree of life. They generally require reference databases that are very large and/or require a large amount of processing to generate. The gap SIANN is designed to fill is when the entire tree of life is irrelevant, and only predefined subsets of organisms need to be detected. For an underlying method we chose read alignment to diagnostic genomic regions because the algorithms for read alignment are highly parallelizable and have been optimized heavily by the community at large (the current implementation of SIANN uses bowtie2 [Langmead & Salzberg, 2012] for the alignment function, but can be adapted to any alignment algorithm). This approach is distinct from using cladespecific marker genes (Segata, et al., 2012) because unique regions that are larger, smaller, or outside of genes can also be used. Furthermore, this approach supports the rapid construction of custom databases using reference genome sets that require only minimal user-supplied structure.',\n",
       "   'cite_spans': [{'start': 39,\n",
       "     'end': 66,\n",
       "     'text': 'Huson, et al., 2011 PhymmBL',\n",
       "     'ref_id': None},\n",
       "    {'start': 138,\n",
       "     'end': 160,\n",
       "     'text': 'Brady & Salzberg, 2011',\n",
       "     'ref_id': None},\n",
       "    {'start': 213, 'end': 225, 'text': 'Liu, et al.,', 'ref_id': None},\n",
       "    {'start': 922,\n",
       "     'end': 949,\n",
       "     'text': '[Langmead & Salzberg, 2012]',\n",
       "     'ref_id': 'BIBREF7'},\n",
       "    {'start': 1090,\n",
       "     'end': 1112,\n",
       "     'text': '(Segata, et al., 2012)',\n",
       "     'ref_id': None}],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'To understand the principle at work, consider a set of reads that have been aligned to the genomes of several strains belonging to two species. Some regions of those genomes are species-specific, some are strain-specific, and some are shared ( Figure  1a ). When a set of reads is aligned to those genomes such that each read is placed in as many locations as it has a match (at a reasonably stringent threshold), visual inspection of the distribution of reads yields an intuitive understanding of the true source organism as Species I/Strain B (Figure 1b ). If Strain B were not present in the reference database, it would still be clear that the organism was an unknown strain of Species I.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 244,\n",
       "     'end': 254,\n",
       "     'text': 'Figure  1a',\n",
       "     'ref_id': None},\n",
       "    {'start': 545, 'end': 555, 'text': '(Figure 1b', 'ref_id': None}],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'The unique identification of a species or strain is quantified by the proportion of the genome that is determined to be species-or strain-specific (defined as reads that are aligned to regions that are species-or strain-specific). Each species and strain is then assigned a numerical measure of the proportion that is covered by these diagnostic reads, and that proportional measure is compared to the ideal case, where sequences from a single organism (generated in silico) are aligned against the database in an identical manner. After that normalization factor is applied, the resulting score indicates whether the source sample contained any of the organisms in the reference database. Figure 1 . A) For a group of strains belonging to two different species, some regions may be unique to each species (region 1), while other regions may be unique to strains within each species (regions 2 and 3). B) A set of reads are aligned to these genomes, and the ones that align in a species-or strain-specific manner are identified by the combination of genomes to which they align. In this example, Strain B of Species I is the organism identified.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 690,\n",
       "     'end': 698,\n",
       "     'text': 'Figure 1',\n",
       "     'ref_id': None}],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'The analysis is conducted independently on both the species and the strain level, so that if the true strain is not present in the database, the species of origin will still be identified. While many methods consider the complete taxonomic tree and assign reads to the least common ancestor, SIANN considers only two taxonomic levels: species and strain, throwing out anything that is not unique at one of those levels and thus obviating many of the confounding factors introduced by manually curated taxonomies.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'The example shown in Figure 1b indicates that species-specific reads are identified as reads that align to one species (Species I, in that case) but not the other. If Species II were not present in the example shown in Figure 1b , a much larger number of reads would be assigned as \"species-specific,\" when in fact those regions are shared with other species. Therefore, the ability of this method to identify strain-and species-specific sequences is a direct function of the inclusion of near neighbors in the reference database. This characteristic is shared among many classification algorithms, but it is of particular note for this method when users have an opportunity to construct their own database.. In order to detect a target species with a high degree of specificity (reducing false positives), it is necessary to include other related species in the reference database. Only by parallel alignment to those near neighbors can the redundant sequences be separated from the species-specific ones. For example, in order to detect Bacillus anthracis in a sample, it would be necessary to include other species of Bacilli in the reference database so that the presence of B. cereus or B. thuringiensis in a sample does not lead to a false call for B. anthracis.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 21, 'end': 30, 'text': 'Figure 1b', 'ref_id': None},\n",
       "    {'start': 219, 'end': 228, 'text': 'Figure 1b', 'ref_id': None}],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'The nomenclature of genus, species, and strain is potentially problematic because it does not correspond to a consistent degree of evolutionary distance or genomic distinctiveness. The ability to distinguish two organisms by any method using genomic sequence data is proportional to the amount of each genome that is shared or unique. One might assume that any two organisms of the same species will have a relatively predictable amount of shared genomic identity. However, some pairs of organisms from the same species may have less in common than other pairs of organisms from different species or even genera. This ambiguity impacts SIANN in two ways. If two organisms have very little genomic sequence to distinguish them, the sensitivity of SIANN to detect either one will diminish (the rate of false negatives will increase as the likelihood of sequencing unique regions decreases). Conversely, if an organism is extremely dissimilar to the near neighbors selected for the database, the specificity with which SIANN detects that organism will decline (the rate of false positives will increase as the number of related genomes available in the database decreases). For example, if a database contained only E. coli and B.anthracis, a sample containing B. cereus would be misidentified as contraining B. anthracis. In the intended use case, a database targeting B. anthracis would contain B. cereus and a number of other near neighbors to prevent that kind of misidentification. It would be convenient to say that an ideal database can be made by calculating the ideal genetic distance between all references and then finding an ideal set of organisms to make up that database, but the behavior of any database will be governed by the particular genomes of the organisms it encounters in the wild. Because not all organisms evolve in the same manner (differences in mutation rate, horizontal gene transfer, recombination, etc), the suitability of a database and method to detect a given organism can only be determined by thorough validation and benchmarking, as well as updating the reference database as needed. Users of SIANN may construct their own custom databases to include newly identified genomes or specific subsets of genomes that best suit their research interests.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'Steps to construct a custom database:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': '1. Select a set of target organisms 2. Gather a set of genome sequences for those target organisms as well as a matched set of near neighbors 3. Using those reference genome sequences as an input, SIANN will: a. Construct a reference index for alignment b. Simulate a set of reads from each genome c. Align each of those simulated read sets to all of the reference genomes d. Calculate the proportion of each reference genome that is strain-or species-specific e. [If two organisms do not have a minimal amount of unique sequences that exceeds the rate of sequencing error, SIANN asks that all but one of those organisms are removed from the database to eliminate redundancy. Note that the user can provide a single representative genome with multiple strain names so that the redundant strain names are not lost.]',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'The files contained within each SIANN database are a compressed genomic index and a list containing the proportion of each reference genome that was found to be strainor species-specific during database construction.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'To run SIANN:',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': '1. Select a pre-made SIANN database and a set of sequences to be analyzed, and 2. SIANN will: a. Align each of the reads against the reference genomes b. Calculate the proportion of each reference genome that is strain-or species-specific within those reads c. Compare that proportion to the simulated ideal case generated during database creation . CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint d. Calculate the probability that the given results could be generated by random chance e. Report the normalized proportion and non-parametric statistic of likelihood for each strain and species in the reference database. The normalized proportion of the genome covered by strain-or speciesspecific reads is the primary statistic reported by this tool.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Name Method Reference MEGAN'},\n",
       "  {'text': 'The performance of SIANN (version 1.6) was tested in comparison to the following stateof-the-art metagenomic classification programs: LMAT (version 1.2), MetaPhlAn (version 1.7.7), and Kraken (version 0.9.1b). All of the programs in Table 1 were investigated for this effort, and three were chosen based on their ability to run on our high-performance computing cluster with an execution time and memory requirement that would be suitable to a clinical lab. Each program was run on a set of 600 simulated datasets generated by MetaSim (Richter, et al., 2008) . Each dataset consisted of 15,000,000 reads (100bp single-ended) with Illumina-simulated error (fourth-degree polynomial) (Korbel, et al., 2009) . The 600 datasets were broken into 12 sets of 50 replicates. Each of the 12 sets contained organisms at different levels of abundance as shown in Table 2 .',\n",
       "   'cite_spans': [{'start': 535,\n",
       "     'end': 558,\n",
       "     'text': '(Richter, et al., 2008)',\n",
       "     'ref_id': 'BIBREF9'},\n",
       "    {'start': 682,\n",
       "     'end': 704,\n",
       "     'text': '(Korbel, et al., 2009)',\n",
       "     'ref_id': 'BIBREF5'}],\n",
       "   'ref_spans': [{'start': 233, 'end': 240, 'text': 'Table 1', 'ref_id': None},\n",
       "    {'start': 852, 'end': 859, 'text': 'Table 2', 'ref_id': None}],\n",
       "   'section': 'Benchmarking'},\n",
       "  {'text': 'Organisms were specifically chosen in pairs so that the ability to distinguish these near neighbors could be determined. The abundances were staggered at 4-fold intervals so that a wide range could be evaluated. All known species of near neighbors for each of the 12 target organisms were included in the reference database used by SIANN for this benchmarking (\"Target Pathogen Database\") and are shown in Appendix 1.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Benchmarking'},\n",
       "  {'text': 'Each program outputs a distinct measure. Kraken and LMAT both count the reads assigned to each taxon, MetaPhlAn calculates the abundance, and SIANN outputs a measure of the proportion of diagnostic genomic regions present. To put these measures on an even footing, we empirically calculated the false positive rate for each method over all 600 samples, at each possible measure of output. Because each dataset is made up of known organisms, any result can be classified as true or false. Therefore, for any possible result (say, 513 reads classified by LMAT or 1.6% abundance assigned by MetaPhlAn), one can calculate the proportion of calls with at least the same amount of support that were correct (True Positives/[True Positives+False Positives]), over all of the 600 datasets. That measure is commonly given as Positive Predictive Value (PPV). For each program, the results can be translated from the raw value into a PPV that is based on this empirical measure of error. The key item of interest is the PPV value for the results that we know to be true positives, the defined spike organisms. Another way of describing this approach is to say that the results of each program have been normalized to the false positive error rate that was empirically observed. If another set of samples were generated, the PPV vs. raw value curve ( Figure 2 ) would likely fall differently, but in this case it gives us a means of comparing a diverse set of methods against the same ground truth. If method 1 detects an organism with a higher PPV than method 2 does, it means that method 1 has fewer false positives in the range that it reports true positives, which is the definition of utility in this setting.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 1339,\n",
       "     'end': 1347,\n",
       "     'text': 'Figure 2',\n",
       "     'ref_id': 'FIGREF0'}],\n",
       "   'section': 'Benchmarking'},\n",
       "  {'text': 'For each method, PPV was calculated as a function of raw output value. Briefly, this was done by compiling the output for all 600 samples, labeling each result as false or true based on the sample set that it came from, and then calculating (at each possible value of output) what the proportion of TP/[TP+FP] was for results with at least that level of raw output. Some simplification steps were taken, such as focusing on the specieslevel assignments (for comparison with methods that do not perform strain assignment), and only taking the top hit for each species from each dataset. Custom R and BASH scripts were used for the data compilation and analysis.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Benchmarking'},\n",
       "  {'text': 'The relationship of raw output value to PPV is shown for each of the four methods in Figure 2 . The point at which PPV is very close to 1 (where 95% of results are true positives) is ~41,000 reads for Kraken, ~2,800 reads for LMAT, ~38% abundance for MetaPhlAn, and 0.21 for SIANN. For SIANN this means that having 38% of the species-unique genome covered by reads resulted in the vast majority of calls being accurate. . CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint For readassignment methods (such as LMAT and Kraken), manual inspection of the results may yield a different understanding of confidence than is presented here, or in any automated analysis.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 85,\n",
       "     'end': 93,\n",
       "     'text': 'Figure 2',\n",
       "     'ref_id': 'FIGREF0'}],\n",
       "   'section': 'Results'},\n",
       "  {'text': 'For example, while each read that is assigned by LMAT and Kraken fall above a certain cutoff for speciesspecificity, some individual reads may be much more specific than others. One could identify a read that aligns to a single species of bacteria with 100% accuracy over its 300bp length, with the next closest match being only 90% similar. It is extremely unlikely that a 300bp exact match would arise due to random chance, and so the user could say with confidence that the organism of interest is found within the sequence data (not considering contamination, horizontal gene transfer, etc). However, such an approach is not currently implemented in an automated method, and many of the steps needed to make that assertion are performed manually by a domain expert, including alignment to near neighbors and ensuring that the read does not fall within a transposon, plasmid, etc. Therefore, while one could say that a single read is all that is needed to state with high PPV that an organism is present, the amount of reads assigned in an automated manner needed to achieve that level of PPV will number in the thousands (Fig 2) . Table 2 ), and each program (boxes at top), across a maximum of 50 replicates (indicated by the size of each point). Note that the reference database for MetaPhlAn does not include viruses, and the reference database for Kraken does not include RNA viruses (e.g., Hanta virus).',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 1125,\n",
       "     'end': 1132,\n",
       "     'text': '(Fig 2)',\n",
       "     'ref_id': 'FIGREF0'},\n",
       "    {'start': 1135, 'end': 1142, 'text': 'Table 2', 'ref_id': None}],\n",
       "   'section': 'Results'},\n",
       "  {'text': '. CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint The next phase of benchmarking was to determine how many raw input reads were needed to achieve the threshold for high PPV. To demonstrate this we plotted the known abundance of each spike organism against the PPV value generated by each method (Figure 3 ). Each point (an organism at a known level of abundance) is comprised of a maximum of 50 replicates, where the diameter of each point increases with an increasing number of replicates. For demonstration purposes we are showing two pairs of bacteria and three viruses. Recall that for each of the pairs of bacteria (and the two poxviruses) any sample containing one did not contain the other (as shown in Table 1 ). The empty boxes result from the organisms not being called at any abundance. For MetaPhlAn, that is a result of no viruses being included in the version of the reference database available for this analysis. Kraken assigned no reads to Hanta virus because viral RNA genomes were not included in this version of the reference database (personal communication with D. Wood). This emphasizes the point that a) the ability to create custom databases targeting organisms of interest can be valuable, and b) the performance of any method must be benchmarked against each potential target of interest.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 460,\n",
       "     'end': 469,\n",
       "     'text': '(Figure 3',\n",
       "     'ref_id': 'FIGREF1'},\n",
       "    {'start': 875, 'end': 882, 'text': 'Table 1', 'ref_id': None}],\n",
       "   'section': 'Results'},\n",
       "  {'text': 'All methods were able identify the bulk of organisms in their databases at high abundances (75% and 18%, Figure 3 ), however performance varied considerably at lower abundances and depended on the particular organism and method used. SIANN detected each organism at high confidence, even at levels as low as 0.3% and 0.07% of the total.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [{'start': 105,\n",
       "     'end': 113,\n",
       "     'text': 'Figure 3',\n",
       "     'ref_id': 'FIGREF1'}],\n",
       "   'section': 'Results'},\n",
       "  {'text': \"The process of detecting trace amounts of a specific organism in a complex mixture of DNA is challenging enough for an expert, but that pales in comparison to the difficulty of accomplishing the same certainty of detection in an automated manner. The results presented here show that SIANN rapidly detects the presence of a given set of organisms with a high degree of specificity and sensitivity. For example, at the 95% confidence (PPV) cutoff of 0.2, SIANN reliably detects all of the organisms tested here at as low as 0.3% abundance. This strong performance is likely due to the fact that SIANN is able to use a method (read alignment to whole genomes) that would be far too computationally costly if it were applied to the entire collection of known genomes. By focusing on a set of (customizable) target organisms and their near neighbors, SIANN can operate quickly and with low computational requirements while delivering highly accurate results. SIANN is available on Illumina's BaseSpace (www.basespace.illumina.com) as a NativeApp, with the database tested here (Appendix 1), as well as a database made from the NCBI representative set of prokaryotic genomes (ftp://ftp.ncbi.nlm.nih.gov/genomes/genome_reports/) (Appendix 2) and the complete set of NCBI viral genomes (ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/) (Appendix 3).\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Discussion'},\n",
       "  {'text': \". CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint BaseSpace was chosen as an appropriate release platform because while the entire set of software and dependencies can be deployed by the user from within a graphical user interface, the actual computation takes place in a controlled 'cloud' environment. Such a distribution strategy obviates the need to satisfy the multiple software or OS dependencies that often arises with academic computational methods. Results for SIANN are compiled into a report format, showing both the organisms that surpass 95% confidence, as well as the closest strain match for each species. The default view masks the raw data output, so that the results are human-readable and do not present extraneous information. While the code for execution and databaseconstruction on a users system is available from Signature Science, LLC, additional databases on the BaseSpace platform can be made available upon request.\",\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Discussion'},\n",
       "  {'text': 'There is a neverending list of questions that one could ask of metagenomic sequencing data generated from important samples. Instead of answering them all, we demonstrate a technique with a very narrow focus that is able to report with a high degree of confidence whether a given set of organisms is present in a sample. These results are presented to the user in a comprehensible format, and accessible on a commonly-used web platform. The world of bioinformatics will continue to progress and develop more sophisticated tools for metagenomic analysis, and we hope that the utility of SIANN will convince others to package and benchmark their tools in a way that they can be used with confidence by the larger public, as well as the research community.',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Discussion'},\n",
       "  {'text': '. CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint',\n",
       "   'cite_spans': [],\n",
       "   'ref_spans': [],\n",
       "   'section': 'Discussion'}],\n",
       " 'bib_entries': {'BIBREF0': {'ref_id': 'b0',\n",
       "   'title': 'Scalable metagenomic taxonomy classification using a reference genome database',\n",
       "   'authors': [{'first': 'S', 'middle': ['K'], 'last': 'Ames', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['A'], 'last': 'Hysom', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['N'], 'last': 'Gardner', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['S'], 'last': 'Lloyd', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['B'], 'last': 'Gokhale', 'suffix': ''},\n",
       "    {'first': 'Allen', 'middle': [], 'last': 'Je', 'suffix': ''}],\n",
       "   'year': 2013,\n",
       "   'venue': 'Bioinformatics',\n",
       "   'volume': '29',\n",
       "   'issn': '18',\n",
       "   'pages': '2253--60',\n",
       "   'other_ids': {'DOI': ['10.1093/bioinformatics/btt389']}},\n",
       "  'BIBREF1': {'ref_id': 'b1',\n",
       "   'title': 'A comparative evaluation of sequence classification programs',\n",
       "   'authors': [{'first': 'A',\n",
       "     'middle': ['L'],\n",
       "     'last': 'Bazinet',\n",
       "     'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['P'], 'last': 'Cummings', 'suffix': ''}],\n",
       "   'year': 2012,\n",
       "   'venue': 'BMC Bioinformatics',\n",
       "   'volume': '13',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {'DOI': ['10.1186/1471-2105-13-92']}},\n",
       "  'BIBREF2': {'ref_id': 'b2',\n",
       "   'title': 'Rapid phylogenetic and functional classification of short genomic fragments with signature peptides',\n",
       "   'authors': [{'first': 'J', 'middle': [], 'last': 'Berendzen', 'suffix': ''},\n",
       "    {'first': 'W', 'middle': ['J'], 'last': 'Bruno', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['D'], 'last': 'Cohn', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': ['W'], 'last': 'Hengartner', 'suffix': ''},\n",
       "    {'first': 'C', 'middle': ['R'], 'last': 'Kuske', 'suffix': ''},\n",
       "    {'first': 'B', 'middle': ['H'], 'last': 'Mcmahon', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['A'], 'last': 'Wolinsky', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': [], 'last': 'Xie', 'suffix': ''}],\n",
       "   'year': 2012,\n",
       "   'venue': 'BMC Research Notes',\n",
       "   'volume': '5',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {'DOI': ['10.1186/1756-0500-5-460']}},\n",
       "  'BIBREF3': {'ref_id': 'b3',\n",
       "   'title': 'Phymm and PhymmBL: metagenomic phylogenetic classification with interpolated Markov models',\n",
       "   'authors': [{'first': 'A', 'middle': [], 'last': 'Brady', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['L'], 'last': 'Salzberg', 'suffix': ''}],\n",
       "   'year': 2009,\n",
       "   'venue': 'Nature Methods',\n",
       "   'volume': '6',\n",
       "   'issn': '9',\n",
       "   'pages': '673--679',\n",
       "   'other_ids': {'DOI': ['10.1038/nmeth.1358']}},\n",
       "  'BIBREF4': {'ref_id': 'b4',\n",
       "   'title': 'Integrative analysis of environmental sequences using MEGAN4',\n",
       "   'authors': [{'first': 'D', 'middle': ['H'], 'last': 'Huson', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Mitra', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Weber', 'suffix': ''},\n",
       "    {'first': 'H', 'middle': [], 'last': 'Ruscheweyh', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['C'], 'last': 'Schuster', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'Genome Research',\n",
       "   'volume': '21',\n",
       "   'issn': '',\n",
       "   'pages': '1552--1560',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF5': {'ref_id': 'b5',\n",
       "   'title': 'PEMer: a computational framework with simulation-based error models for inferring genomic structural variants from massive paired-end sequencing data',\n",
       "   'authors': [{'first': 'J', 'middle': ['O'], 'last': 'Korbel', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Abyzov', 'suffix': ''},\n",
       "    {'first': 'X', 'middle': ['J'], 'last': 'Mu', 'suffix': ''},\n",
       "    {'first': 'N', 'middle': [], 'last': 'Carriero', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Cayting', 'suffix': ''},\n",
       "    {'first': 'Z', 'middle': [], 'last': 'Zhang', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Snyder', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': ['B'], 'last': 'Gerstein', 'suffix': ''}],\n",
       "   'year': 2009,\n",
       "   'venue': '',\n",
       "   'volume': '',\n",
       "   'issn': '',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF7': {'ref_id': 'b7',\n",
       "   'title': 'Fast gapped-read alignment with Bowtie 2',\n",
       "   'authors': [{'first': 'B', 'middle': [], 'last': 'Langmead', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Salzberg', 'suffix': ''}],\n",
       "   'year': 2012,\n",
       "   'venue': 'Nature Methods',\n",
       "   'volume': '9',\n",
       "   'issn': '',\n",
       "   'pages': '357--359',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF8': {'ref_id': 'b8',\n",
       "   'title': 'Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences',\n",
       "   'authors': [{'first': 'B', 'middle': [], 'last': 'Liu', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': [], 'last': 'Gibbons', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Ghodsi', 'suffix': ''},\n",
       "    {'first': 'T', 'middle': [], 'last': 'Treangen', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Pop', 'suffix': ''}],\n",
       "   'year': 2011,\n",
       "   'venue': 'BMC Genomics',\n",
       "   'volume': '12',\n",
       "   'issn': '2',\n",
       "   'pages': '',\n",
       "   'other_ids': {}},\n",
       "  'BIBREF9': {'ref_id': 'b9',\n",
       "   'title': 'MetaSim: a sequencing simulator for genomics and metagenomics',\n",
       "   'authors': [{'first': 'D',\n",
       "     'middle': ['C'],\n",
       "     'last': 'Richter',\n",
       "     'suffix': ''},\n",
       "    {'first': 'F', 'middle': [], 'last': 'Ott', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': ['F'], 'last': 'Auch', 'suffix': ''},\n",
       "    {'first': 'R', 'middle': [], 'last': 'Schmid', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['H'], 'last': 'Huson', 'suffix': ''}],\n",
       "   'year': 2008,\n",
       "   'venue': 'PLoS One',\n",
       "   'volume': '3',\n",
       "   'issn': '10',\n",
       "   'pages': '',\n",
       "   'other_ids': {'DOI': ['10.1371/journal.pone.0003373']}},\n",
       "  'BIBREF10': {'ref_id': 'b10',\n",
       "   'title': 'Genomic variation landscape of the human gut microbiome',\n",
       "   'authors': [{'first': 'S',\n",
       "     'middle': [],\n",
       "     'last': 'Schloissnig',\n",
       "     'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Arumugam', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': [], 'last': 'Sunagawa', 'suffix': ''},\n",
       "    {'first': 'M', 'middle': [], 'last': 'Mitreva', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Tap', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Zhu', 'suffix': ''},\n",
       "    {'first': 'A', 'middle': [], 'last': 'Waller', 'suffix': ''},\n",
       "    {'first': 'D', 'middle': ['R'], 'last': 'Mende', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': ['R'], 'last': 'Kultima', 'suffix': ''},\n",
       "    {'first': 'J', 'middle': [], 'last': 'Martin', 'suffix': ''},\n",
       "    {'first': 'K', 'middle': [], 'last': 'Kota', 'suffix': ''},\n",
       "    {'first': 'S', 'middle': ['R'], 'last': 'Sunyaev', 'suffix': ''},\n",
       "    {'first': 'G', 'middle': ['M'], 'last': 'Weinstock', 'suffix': ''},\n",
       "    {'first': 'P', 'middle': [], 'last': 'Bork', 'suffix': ''}],\n",
       "   'year': 2013,\n",
       "   'venue': 'Nature',\n",
       "   'volume': '493',\n",
       "   'issn': '7430',\n",
       "   'pages': '45--50',\n",
       "   'other_ids': {'DOI': ['10.1038/nature11711']}}},\n",
       " 'ref_entries': {'FIGREF0': {'text': 'Relationship of reported value for each program (horizontal axis, log scale) to the empirically-determined Positive Predictive Value (PPV), shown on the vertical axis. While the exact values depend on the test data used, the general values at significant cutoff values (0.8, 0.9, 0.95 PPV) remain relatively constant across different datasets (data not shown).',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'FIGREF1': {'text': 'The Positive Predictive Value (PPV, vertical axis) is shown for each organism (boxes on right), at each level of known abundance (horizontal axis, see',\n",
       "   'latex': None,\n",
       "   'type': 'figure'},\n",
       "  'TABREF0': {'text': '. Summary of methods for metagenomic classification.',\n",
       "   'latex': None,\n",
       "   'type': 'table'},\n",
       "  'TABREF1': {'text': 'Table 2. The abundance of each target organism in each set of simulated datasets. Each set is indicated by the number in the top row, and was generated with 50 replicates.',\n",
       "   'latex': None,\n",
       "   'type': 'table'},\n",
       "  'TABREF2': {'text': 'Segata N, Waldron L, Ballarini A, Narasimhan V,Jousson O, and Huttenhower C (2012). Metagenomic microbial community profiling using unique clade-specific marker genes. NatureMethods, 9(8):811-4. doi: 10.1038/nmeth.2066. Sunagawa S, Mende DR, Zeller G, Izquierdo-Carrasco F, Berger SA, Kultima JR, Coelho LP, Arumugam M, Tap J, Nielsen HB, Rasmussen S, Brunak S, Pedersen O, Guarner F, de Vos WM, Wang J, Li J, Dor J, Ehrlich SD, Stamatakis A & Bork P (2013). Metagenomic species profiling using universal phylogenetic marker genes. Nature Methods, 10, 1196-1199 (2013) doi:10.1038/nmeth.2693 Wood DE and Salzberg SL. Ultrafast metagenomic sequence classification using exact alignments. In submission. Wu GD, Chen J, Hoffmann C, Bittinger K, Chen YY, Keilbaugh SA, Bewtra M, Knights D, Walters WA, Knight R, Sinha R, Gilroy E, Gupta K, Baldassano R, Nessel L, Li H, Bushman FD, Lewis JD (2011). Linking long-term dietary patterns with gut microbial enterotypes. Science. 334(6052):105-8. doi: 10.1126/science.1208344. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint Appendix 2: Viral Database Abaca bunchy top virus DNA-C Abaca bunchy top virus DNA-M Abaca bunchy top virus DNA-The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101Circovirus-like genome BBC-A Circovirus-like genome CB-A Circovirus-like genome CB-B Circovirus-like genome RW-A Circovirus-like genome RW-B Circovirus-like genome RW-C Circovirus-like genome RW-D Circovirus-like genome RW-E Circovirus-like genome SAR-A Circovirus-like genome SAR-B Citrus psorosis virus RNA1 Citrus psorosis virus RNA2 Citrus psorosis virus RNA3The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprintThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprintThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint . CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint . CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder.It . https://doi.org/10.1101/001727 doi: bioRxiv preprint Thermovirga lienii DSM 17291 . CC-BY-NC-ND 4.0 International license is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. It . https://doi.org/10.1101/001727 doi: bioRxiv preprint Thermus scotoductus SA 01 Thermus sp CCB US3 UF1 Thermus thermophilus HB27 Vibrio cholerae O1 biovar El Tor str N16961 Vibrio fischeri ES114 Vibrio furnissii NCTC 11218 Vibrio harveyi ATCC BAA 1116 . CC-BY-NC-ND 4.0 International license is made available under a',\n",
       "   'latex': None,\n",
       "   'type': 'table'}},\n",
       " 'back_matter': []}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'abstract': 'To study the genetic diversity of enterovirus G (EV-G) among Japanese pigs, metagenomics sequencing was performed on fecal samples from pigs with or without diarrhea, collected between 2014 and 2016. Fifty-nine EV-G sequences, which were >5,000 nucleotides long, were obtained. By complete VP1 sequence analysis, Japanese EV-G isolates were classified into G1 (17 strains), G2 (four strains), G3 (22 strains), G4 (two strains), G6 (two strains), G9 (six strains), G10 (five strains), and a new genotype (one strain). Remarkably, 16 G1 and one G2 strain identified in diarrheic (23.5%; four strains) or normal (76.5%; 13 strains) fecal samples possessed a papain-like cysteine protease (PL-CP) sequence, which was recently found in the USA and Belgium in the EV-G genome, at the 2C-3A junction site. This paper presents the first report of the high prevalence of viruses carrying PL-CP in the EV-G population. Furthermore, possible inter- and intragenotype recombination events were found among EV-G strains, including G1-PL-CP strains. Our findings may advance the understanding of the molecular epidemiology and genetic evolution of EV-Gs.',\n",
       " 'arxivId': None,\n",
       " 'authors': [{'authorId': '5726175',\n",
       "   'name': 'Shinobu Tsuchiaka',\n",
       "   'url': 'https://www.semanticscholar.org/author/5726175'},\n",
       "  {'authorId': '6258941',\n",
       "   'name': 'Yuki Naoi',\n",
       "   'url': 'https://www.semanticscholar.org/author/6258941'},\n",
       "  {'authorId': '48910962',\n",
       "   'name': 'Ryo Imai',\n",
       "   'url': 'https://www.semanticscholar.org/author/48910962'},\n",
       "  {'authorId': '4002097',\n",
       "   'name': 'Tsuneyuki Masuda',\n",
       "   'url': 'https://www.semanticscholar.org/author/4002097'},\n",
       "  {'authorId': '47209124',\n",
       "   'name': 'Mika Ito',\n",
       "   'url': 'https://www.semanticscholar.org/author/47209124'},\n",
       "  {'authorId': '10682727',\n",
       "   'name': 'Masataka Akagami',\n",
       "   'url': 'https://www.semanticscholar.org/author/10682727'},\n",
       "  {'authorId': '46439640',\n",
       "   'name': 'Yoshinao Ouchi',\n",
       "   'url': 'https://www.semanticscholar.org/author/46439640'},\n",
       "  {'authorId': '49584227',\n",
       "   'name': 'Kazuo Ishii',\n",
       "   'url': 'https://www.semanticscholar.org/author/49584227'},\n",
       "  {'authorId': '48706772',\n",
       "   'name': 'Shoichi Sakaguchi',\n",
       "   'url': 'https://www.semanticscholar.org/author/48706772'},\n",
       "  {'authorId': '4751067',\n",
       "   'name': 'Tsutomu Omatsu',\n",
       "   'url': 'https://www.semanticscholar.org/author/4751067'},\n",
       "  {'authorId': '36375266',\n",
       "   'name': 'Yukie Katayama',\n",
       "   'url': 'https://www.semanticscholar.org/author/36375266'},\n",
       "  {'authorId': '5144887',\n",
       "   'name': 'Mami Oba',\n",
       "   'url': 'https://www.semanticscholar.org/author/5144887'},\n",
       "  {'authorId': '4511770',\n",
       "   'name': 'Junsuke Shirai',\n",
       "   'url': 'https://www.semanticscholar.org/author/4511770'},\n",
       "  {'authorId': '50004571',\n",
       "   'name': 'Yuki Satani',\n",
       "   'url': 'https://www.semanticscholar.org/author/50004571'},\n",
       "  {'authorId': '1794862',\n",
       "   'name': 'Yasuhiro Takashima',\n",
       "   'url': 'https://www.semanticscholar.org/author/1794862'},\n",
       "  {'authorId': '47249257',\n",
       "   'name': 'Yuji Taniguchi',\n",
       "   'url': 'https://www.semanticscholar.org/author/47249257'},\n",
       "  {'authorId': '5799180',\n",
       "   'name': 'Masaki Takasu',\n",
       "   'url': 'https://www.semanticscholar.org/author/5799180'},\n",
       "  {'authorId': '2489469',\n",
       "   'name': 'Hiroo Madarame',\n",
       "   'url': 'https://www.semanticscholar.org/author/2489469'},\n",
       "  {'authorId': '2053913',\n",
       "   'name': 'Fujiko Sunaga',\n",
       "   'url': 'https://www.semanticscholar.org/author/2053913'},\n",
       "  {'authorId': '34762043',\n",
       "   'name': 'Hiroshi Aoki',\n",
       "   'url': 'https://www.semanticscholar.org/author/34762043'},\n",
       "  {'authorId': '2358043',\n",
       "   'name': 'Shinji Makino',\n",
       "   'url': 'https://www.semanticscholar.org/author/2358043'},\n",
       "  {'authorId': '49359835',\n",
       "   'name': 'Tetsuya Mizutani',\n",
       "   'url': 'https://www.semanticscholar.org/author/49359835'},\n",
       "  {'authorId': '2993964',\n",
       "   'name': 'Makoto Nagai',\n",
       "   'url': 'https://www.semanticscholar.org/author/2993964'}],\n",
       " 'citationVelocity': 0,\n",
       " 'citations': [{'arxivId': None,\n",
       "   'authors': [{'authorId': '49220236',\n",
       "     'name': 'Yuki Fujii',\n",
       "     'url': 'https://www.semanticscholar.org/author/49220236'},\n",
       "    {'authorId': '35657384',\n",
       "     'name': 'Yuki Kashima',\n",
       "     'url': 'https://www.semanticscholar.org/author/35657384'},\n",
       "    {'authorId': '50371912',\n",
       "     'name': 'Fujiko Sunaga',\n",
       "     'url': 'https://www.semanticscholar.org/author/50371912'},\n",
       "    {'authorId': '34762043',\n",
       "     'name': 'Hiroshi Aoki',\n",
       "     'url': 'https://www.semanticscholar.org/author/34762043'},\n",
       "    {'authorId': '48910919',\n",
       "     'name': 'Ryo Imai',\n",
       "     'url': 'https://www.semanticscholar.org/author/48910919'},\n",
       "    {'authorId': '6934164',\n",
       "     'name': 'Kaori Sano',\n",
       "     'url': 'https://www.semanticscholar.org/author/6934164'},\n",
       "    {'authorId': '36375266',\n",
       "     'name': 'Yukie Katayama',\n",
       "     'url': 'https://www.semanticscholar.org/author/36375266'},\n",
       "    {'authorId': '4751067',\n",
       "     'name': 'Tsutomu Omatsu',\n",
       "     'url': 'https://www.semanticscholar.org/author/4751067'},\n",
       "    {'authorId': '5144887',\n",
       "     'name': 'Mami Oba',\n",
       "     'url': 'https://www.semanticscholar.org/author/5144887'},\n",
       "    {'authorId': '6199928',\n",
       "     'name': 'Tetsuya Furuya',\n",
       "     'url': 'https://www.semanticscholar.org/author/6199928'},\n",
       "    {'authorId': '48649287',\n",
       "     'name': 'Satoko Tsuzuku',\n",
       "     'url': 'https://www.semanticscholar.org/author/48649287'},\n",
       "    {'authorId': '46439640',\n",
       "     'name': 'Yoshinao Ouchi',\n",
       "     'url': 'https://www.semanticscholar.org/author/46439640'},\n",
       "    {'authorId': '4511770',\n",
       "     'name': 'Junsuke Shirai',\n",
       "     'url': 'https://www.semanticscholar.org/author/4511770'},\n",
       "    {'authorId': '49359835',\n",
       "     'name': 'Tetsuya Mizutani',\n",
       "     'url': 'https://www.semanticscholar.org/author/49359835'},\n",
       "    {'authorId': '34127765',\n",
       "     'name': 'Tomoichiro Oka',\n",
       "     'url': 'https://www.semanticscholar.org/author/34127765'},\n",
       "    {'authorId': '2993964',\n",
       "     'name': 'Makoto Nagai',\n",
       "     'url': 'https://www.semanticscholar.org/author/2993964'}],\n",
       "   'doi': '10.1007/s00705-019-04514-4',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'bef4a594067605106890278f749b1147165cc9bc',\n",
       "   'title': 'Complete genome sequencing and genetic analysis of a Japanese porcine torovirus strain detected in swine feces',\n",
       "   'url': 'https://www.semanticscholar.org/paper/bef4a594067605106890278f749b1147165cc9bc',\n",
       "   'venue': 'Archives of Virology',\n",
       "   'year': 2019},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '102723857',\n",
       "     'name': 'Zhangmin Hu',\n",
       "     'url': 'https://www.semanticscholar.org/author/102723857'},\n",
       "    {'authorId': '49307544',\n",
       "     'name': 'Yong-le Yang',\n",
       "     'url': 'https://www.semanticscholar.org/author/49307544'},\n",
       "    {'authorId': '47775727',\n",
       "     'name': 'Ling-Dong Xu',\n",
       "     'url': 'https://www.semanticscholar.org/author/47775727'},\n",
       "    {'authorId': '144575036',\n",
       "     'name': 'Bin Wang',\n",
       "     'url': 'https://www.semanticscholar.org/author/144575036'},\n",
       "    {'authorId': '48699221',\n",
       "     'name': 'Pan Qin',\n",
       "     'url': 'https://www.semanticscholar.org/author/48699221'},\n",
       "    {'authorId': '48355744',\n",
       "     'name': 'Yao-Wei Huang',\n",
       "     'url': 'https://www.semanticscholar.org/author/48355744'}],\n",
       "   'doi': '10.3389/fvets.2019.00120',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '89a70f5a269900d9e7e4a8b2e325ea1b4adf1bb5',\n",
       "   'title': 'Porcine Torovirus (PToV)A Brief Review of Etiology, Diagnostic Assays and Current Epidemiology',\n",
       "   'url': 'https://www.semanticscholar.org/paper/89a70f5a269900d9e7e4a8b2e325ea1b4adf1bb5',\n",
       "   'venue': 'Front. Vet. Sci.',\n",
       "   'year': 2019},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '2053913',\n",
       "     'name': 'Fujiko Sunaga',\n",
       "     'url': 'https://www.semanticscholar.org/author/2053913'},\n",
       "    {'authorId': '4002097',\n",
       "     'name': 'Tsuneyuki Masuda',\n",
       "     'url': 'https://www.semanticscholar.org/author/4002097'},\n",
       "    {'authorId': '47209124',\n",
       "     'name': 'Mika Ito',\n",
       "     'url': 'https://www.semanticscholar.org/author/47209124'},\n",
       "    {'authorId': '10682727',\n",
       "     'name': 'Masataka Akagami',\n",
       "     'url': 'https://www.semanticscholar.org/author/10682727'},\n",
       "    {'authorId': '6258941',\n",
       "     'name': 'Yuki Naoi',\n",
       "     'url': 'https://www.semanticscholar.org/author/6258941'},\n",
       "    {'authorId': '6934164',\n",
       "     'name': 'Kaori Sano',\n",
       "     'url': 'https://www.semanticscholar.org/author/6934164'},\n",
       "    {'authorId': '36375266',\n",
       "     'name': 'Yukie Katayama',\n",
       "     'url': 'https://www.semanticscholar.org/author/36375266'},\n",
       "    {'authorId': '4751067',\n",
       "     'name': 'Tsutomu Omatsu',\n",
       "     'url': 'https://www.semanticscholar.org/author/4751067'},\n",
       "    {'authorId': '5144887',\n",
       "     'name': 'Mami Oba',\n",
       "     'url': 'https://www.semanticscholar.org/author/5144887'},\n",
       "    {'authorId': '48706772',\n",
       "     'name': 'Shoichi Sakaguchi',\n",
       "     'url': 'https://www.semanticscholar.org/author/48706772'},\n",
       "    {'authorId': '6199928',\n",
       "     'name': 'Tetsuya Furuya',\n",
       "     'url': 'https://www.semanticscholar.org/author/6199928'},\n",
       "    {'authorId': '5656865',\n",
       "     'name': 'Hiroshi Yamasato',\n",
       "     'url': 'https://www.semanticscholar.org/author/5656865'},\n",
       "    {'authorId': '46439640',\n",
       "     'name': 'Yoshinao Ouchi',\n",
       "     'url': 'https://www.semanticscholar.org/author/46439640'},\n",
       "    {'authorId': '4511770',\n",
       "     'name': 'Junsuke Shirai',\n",
       "     'url': 'https://www.semanticscholar.org/author/4511770'},\n",
       "    {'authorId': '49359835',\n",
       "     'name': 'Tetsuya Mizutani',\n",
       "     'url': 'https://www.semanticscholar.org/author/49359835'},\n",
       "    {'authorId': '2993964',\n",
       "     'name': 'Makoto Nagai',\n",
       "     'url': 'https://www.semanticscholar.org/author/2993964'}],\n",
       "   'doi': '10.1007/s11262-019-01640-8',\n",
       "   'intent': ['methodology', 'background'],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'bf9f5d4f2c1b3c43753a1af8133c30611288e463',\n",
       "   'title': 'Complete genomic analysis and molecular characterization of Japanese porcine sapeloviruses',\n",
       "   'url': 'https://www.semanticscholar.org/paper/bf9f5d4f2c1b3c43753a1af8133c30611288e463',\n",
       "   'venue': 'Virus Genes',\n",
       "   'year': 2019},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5409759',\n",
       "     'name': 'Sebastiaan Theuns',\n",
       "     'url': 'https://www.semanticscholar.org/author/5409759'},\n",
       "    {'authorId': '14979913',\n",
       "     'name': 'Bert Vanmechelen',\n",
       "     'url': 'https://www.semanticscholar.org/author/14979913'},\n",
       "    {'authorId': '51006344',\n",
       "     'name': 'Quinten Bernaert',\n",
       "     'url': 'https://www.semanticscholar.org/author/51006344'},\n",
       "    {'authorId': '3769979',\n",
       "     'name': 'Ward Deboutte',\n",
       "     'url': 'https://www.semanticscholar.org/author/3769979'},\n",
       "    {'authorId': '51020602',\n",
       "     'name': 'Marilou Vandenhole',\n",
       "     'url': 'https://www.semanticscholar.org/author/51020602'},\n",
       "    {'authorId': '48471024',\n",
       "     'name': 'Leen Beller',\n",
       "     'url': 'https://www.semanticscholar.org/author/48471024'},\n",
       "    {'authorId': '145734314',\n",
       "     'name': 'Jelle Matthijnssens',\n",
       "     'url': 'https://www.semanticscholar.org/author/145734314'},\n",
       "    {'authorId': '9332657',\n",
       "     'name': 'Piet Maes',\n",
       "     'url': 'https://www.semanticscholar.org/author/9332657'},\n",
       "    {'authorId': '6750977',\n",
       "     'name': 'Hans J. Nauwynck',\n",
       "     'url': 'https://www.semanticscholar.org/author/6750977'}],\n",
       "   'doi': '10.1038/s41598-018-28180-9',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'eff8bed68ef6109e8f0c51a8b1ec4b6ca5b6329e',\n",
       "   'title': 'Nanopore sequencing as a revolutionary diagnostic tool for porcine viral enteric disease complexes identifies porcine kobuvirus as an important enteric virus',\n",
       "   'url': 'https://www.semanticscholar.org/paper/eff8bed68ef6109e8f0c51a8b1ec4b6ca5b6329e',\n",
       "   'venue': 'Scientific Reports',\n",
       "   'year': 2018},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6522233',\n",
       "     'name': 'Claire Muslin',\n",
       "     'url': 'https://www.semanticscholar.org/author/6522233'},\n",
       "    {'authorId': '1382480302',\n",
       "     'name': 'Alice Mac Kain',\n",
       "     'url': 'https://www.semanticscholar.org/author/1382480302'},\n",
       "    {'authorId': '1381441610',\n",
       "     'name': 'Mal Bessaud',\n",
       "     'url': 'https://www.semanticscholar.org/author/1381441610'},\n",
       "    {'authorId': '1382480835',\n",
       "     'name': 'Bruno Blondel',\n",
       "     'url': 'https://www.semanticscholar.org/author/1382480835'},\n",
       "    {'authorId': '47015067',\n",
       "     'name': 'Franis Delpeyroux',\n",
       "     'url': 'https://www.semanticscholar.org/author/47015067'}],\n",
       "   'doi': '10.3390/v11090859',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'ae1f275b97c0306e3167f73613fec9a704746eba',\n",
       "   'title': 'Recombination in Enteroviruses, a Multi-Step Modular Evolutionary Process',\n",
       "   'url': 'https://www.semanticscholar.org/paper/ae1f275b97c0306e3167f73613fec9a704746eba',\n",
       "   'venue': 'Viruses',\n",
       "   'year': 2019},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '1490695588',\n",
       "     'name': 'Kirsty T T Kwok',\n",
       "     'url': 'https://www.semanticscholar.org/author/1490695588'},\n",
       "    {'authorId': '4858713',\n",
       "     'name': 'David F. Nieuwenhuijse',\n",
       "     'url': 'https://www.semanticscholar.org/author/4858713'},\n",
       "    {'authorId': '39367219',\n",
       "     'name': 'My V T Phan',\n",
       "     'url': 'https://www.semanticscholar.org/author/39367219'},\n",
       "    {'authorId': '145963941',\n",
       "     'name': 'Marion P G Koopmans',\n",
       "     'url': 'https://www.semanticscholar.org/author/145963941'}],\n",
       "   'doi': '10.3390/v12010107',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '63f58030dbe7e46d82ccf7794885cac6a912351f',\n",
       "   'title': 'Virus Metagenomics in Farm Animals: A Systematic Review',\n",
       "   'url': 'https://www.semanticscholar.org/paper/63f58030dbe7e46d82ccf7794885cac6a912351f',\n",
       "   'venue': 'Viruses',\n",
       "   'year': 2020}],\n",
       " 'corpusId': 19710097,\n",
       " 'doi': '10.1371/journal.pone.0190819',\n",
       " 'fieldsOfStudy': ['Medicine', 'Biology'],\n",
       " 'influentialCitationCount': 0,\n",
       " 'is_open_access': True,\n",
       " 'is_publisher_licensed': True,\n",
       " 'paperId': 'eceb1667ac6722e9131eb7d0f560a7f76bfb25f1',\n",
       " 'references': [{'arxivId': None,\n",
       "   'authors': [{'authorId': '1419294346',\n",
       "     'name': 'Sophia Austermann-Busch',\n",
       "     'url': 'https://www.semanticscholar.org/author/1419294346'},\n",
       "    {'authorId': '5039656',\n",
       "     'name': 'Paul Becher',\n",
       "     'url': 'https://www.semanticscholar.org/author/5039656'}],\n",
       "   'doi': '10.1128/JVI.00864-12',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '95bd0a2940b666320f5f4daf02f2ef7d393bf201',\n",
       "   'title': 'RNA structural elements determine frequency and sites of nonhomologous recombination in an animal plus-strand RNA virus.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/95bd0a2940b666320f5f4daf02f2ef7d393bf201',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2012},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '4885578',\n",
       "     'name': 'Brian Skaug',\n",
       "     'url': 'https://www.semanticscholar.org/author/4885578'},\n",
       "    {'authorId': '38308052',\n",
       "     'name': 'Z. Jeffrey Chen',\n",
       "     'url': 'https://www.semanticscholar.org/author/38308052'}],\n",
       "   'doi': '10.1016/j.cell.2010.09.033',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'bf0c400c78c83f1eae4317aed849211d965cfc46',\n",
       "   'title': 'Emerging Role of ISG15 in Antiviral Immunity',\n",
       "   'url': 'https://www.semanticscholar.org/paper/bf0c400c78c83f1eae4317aed849211d965cfc46',\n",
       "   'venue': 'Cell',\n",
       "   'year': 2010},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '144215561',\n",
       "     'name': 'kos Boros',\n",
       "     'url': 'https://www.semanticscholar.org/author/144215561'},\n",
       "    {'authorId': '5601977',\n",
       "     'name': 'Pter Pankovics',\n",
       "     'url': 'https://www.semanticscholar.org/author/5601977'},\n",
       "    {'authorId': '88574239',\n",
       "     'name': 'Nick J. Knowles',\n",
       "     'url': 'https://www.semanticscholar.org/author/88574239'},\n",
       "    {'authorId': '8964093',\n",
       "     'name': 'Gbor Reuter',\n",
       "     'url': 'https://www.semanticscholar.org/author/8964093'}],\n",
       "   'doi': '10.1099/vir.0.041335-0',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '03b8b149e62f330c7a959e13c563867de1694a31',\n",
       "   'title': 'Natural interspecies recombinant bovine/porcine enterovirus in sheep.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/03b8b149e62f330c7a959e13c563867de1694a31',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2012},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '144215561',\n",
       "     'name': 'kos Boros',\n",
       "     'url': 'https://www.semanticscholar.org/author/144215561'},\n",
       "    {'authorId': '35154591',\n",
       "     'name': 'Csaba Nemes',\n",
       "     'url': 'https://www.semanticscholar.org/author/35154591'},\n",
       "    {'authorId': '5601977',\n",
       "     'name': 'Pter Pankovics',\n",
       "     'url': 'https://www.semanticscholar.org/author/5601977'},\n",
       "    {'authorId': '4275699',\n",
       "     'name': 'Hunor Br',\n",
       "     'url': 'https://www.semanticscholar.org/author/4275699'},\n",
       "    {'authorId': '5046835',\n",
       "     'name': 'Beatrix Kapusinszky',\n",
       "     'url': 'https://www.semanticscholar.org/author/5046835'},\n",
       "    {'authorId': '4384073',\n",
       "     'name': 'Eric L Delwart',\n",
       "     'url': 'https://www.semanticscholar.org/author/4384073'},\n",
       "    {'authorId': '8964093',\n",
       "     'name': 'Gbor Reuter',\n",
       "     'url': 'https://www.semanticscholar.org/author/8964093'}],\n",
       "   'doi': '10.1007/s00705-012-1255-5',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '8c699b3b6e0238d727a3469511f789e6a03da377',\n",
       "   'title': 'Characterization of a novel porcine enterovirus in wild boars in Hungary',\n",
       "   'url': 'https://www.semanticscholar.org/paper/8c699b3b6e0238d727a3469511f789e6a03da377',\n",
       "   'venue': 'Archives of Virology',\n",
       "   'year': 2012},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '48878999',\n",
       "     'name': 'Shu-Chun Chiu',\n",
       "     'url': 'https://www.semanticscholar.org/author/48878999'},\n",
       "    {'authorId': '48473415',\n",
       "     'name': 'Chih-Lin Yang',\n",
       "     'url': 'https://www.semanticscholar.org/author/48473415'},\n",
       "    {'authorId': '4952674',\n",
       "     'name': 'Ya-Mei Chen',\n",
       "     'url': 'https://www.semanticscholar.org/author/4952674'},\n",
       "    {'authorId': '15789453',\n",
       "     'name': 'Shu-Chia Hu',\n",
       "     'url': 'https://www.semanticscholar.org/author/15789453'},\n",
       "    {'authorId': '12085380',\n",
       "     'name': 'Kuo-Chao Chiu',\n",
       "     'url': 'https://www.semanticscholar.org/author/12085380'},\n",
       "    {'authorId': '46393429',\n",
       "     'name': 'Yi-Chien Lin',\n",
       "     'url': 'https://www.semanticscholar.org/author/46393429'},\n",
       "    {'authorId': '2782224',\n",
       "     'name': 'Chia-Yi Chang',\n",
       "     'url': 'https://www.semanticscholar.org/author/2782224'},\n",
       "    {'authorId': '88985721',\n",
       "     'name': 'F I Wang',\n",
       "     'url': 'https://www.semanticscholar.org/author/88985721'}],\n",
       "   'doi': '10.1016/j.vetmic.2013.10.019',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'aca7f552b99686421dc5732fa078c42f1059d032',\n",
       "   'title': 'Multiple models of porcine teschovirus pathogenesis in endemically infected pigs.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/aca7f552b99686421dc5732fa078c42f1059d032',\n",
       "   'venue': 'Veterinary microbiology',\n",
       "   'year': 2014},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5106969',\n",
       "     'name': 'Kavita S. Lole',\n",
       "     'url': 'https://www.semanticscholar.org/author/5106969'},\n",
       "    {'authorId': '35018833',\n",
       "     'name': 'Richard C. Bollinger',\n",
       "     'url': 'https://www.semanticscholar.org/author/35018833'},\n",
       "    {'authorId': '3617485',\n",
       "     'name': 'Ramesh Shivram Paranjape',\n",
       "     'url': 'https://www.semanticscholar.org/author/3617485'},\n",
       "    {'authorId': '153895750',\n",
       "     'name': 'D Gadkari',\n",
       "     'url': 'https://www.semanticscholar.org/author/153895750'},\n",
       "    {'authorId': '3543698',\n",
       "     'name': 'Smita Kulkarni',\n",
       "     'url': 'https://www.semanticscholar.org/author/3543698'},\n",
       "    {'authorId': '38167421',\n",
       "     'name': 'Nicole G. Novak',\n",
       "     'url': 'https://www.semanticscholar.org/author/38167421'},\n",
       "    {'authorId': '3366563',\n",
       "     'name': 'Roxann Ingersoll',\n",
       "     'url': 'https://www.semanticscholar.org/author/3366563'},\n",
       "    {'authorId': '145402543',\n",
       "     'name': 'Haynes W. Sheppard',\n",
       "     'url': 'https://www.semanticscholar.org/author/145402543'},\n",
       "    {'authorId': '32347943',\n",
       "     'name': 'Stuart C. Ray',\n",
       "     'url': 'https://www.semanticscholar.org/author/32347943'}],\n",
       "   'doi': '10.1128/jvi.73.1.152-160.1999',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '30d19c81318b9c0db8ab9564b62472f6e1a4ab2a',\n",
       "   'title': 'Full-length human immunodeficiency virus type 1 genomes from subtype C-infected seroconverters in India, with evidence of intersubtype recombination.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/30d19c81318b9c0db8ab9564b62472f6e1a4ab2a',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 1999},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6425126',\n",
       "     'name': 'Shixing Yang',\n",
       "     'url': 'https://www.semanticscholar.org/author/6425126'},\n",
       "    {'authorId': '46395774',\n",
       "     'name': 'Yan Wang',\n",
       "     'url': 'https://www.semanticscholar.org/author/46395774'},\n",
       "    {'authorId': '39906629',\n",
       "     'name': 'Quan Hu Shen',\n",
       "     'url': 'https://www.semanticscholar.org/author/39906629'},\n",
       "    {'authorId': '2151884',\n",
       "     'name': 'Wen Zhang',\n",
       "     'url': 'https://www.semanticscholar.org/author/2151884'},\n",
       "    {'authorId': '2154634',\n",
       "     'name': 'Xiuguo Hua',\n",
       "     'url': 'https://www.semanticscholar.org/author/2154634'}],\n",
       "   'doi': '10.1186/1743-422X-10-99',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '57081cd91a8e02a6d5a2cb949d1a72dcca0294ac',\n",
       "   'title': 'Prevalence of porcine enterovirus 9 in pigs in Middle and Eastern China',\n",
       "   'url': 'https://www.semanticscholar.org/paper/57081cd91a8e02a6d5a2cb949d1a72dcca0294ac',\n",
       "   'venue': 'Virology Journal',\n",
       "   'year': 2011},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '1996001',\n",
       "     'name': 'Manabu Yamada',\n",
       "     'url': 'https://www.semanticscholar.org/author/1996001'},\n",
       "    {'authorId': '5326856',\n",
       "     'name': 'Ayako Miyazaki',\n",
       "     'url': 'https://www.semanticscholar.org/author/5326856'},\n",
       "    {'authorId': '145531415',\n",
       "     'name': 'Yu Yamamoto',\n",
       "     'url': 'https://www.semanticscholar.org/author/145531415'},\n",
       "    {'authorId': '48569450',\n",
       "     'name': 'Kikuyasu Nakamura',\n",
       "     'url': 'https://www.semanticscholar.org/author/48569450'},\n",
       "    {'authorId': '50230979',\n",
       "     'name': 'Mitsuru Ito',\n",
       "     'url': 'https://www.semanticscholar.org/author/50230979'},\n",
       "    {'authorId': '3807223',\n",
       "     'name': 'Hiroshi Tsunemitsu',\n",
       "     'url': 'https://www.semanticscholar.org/author/3807223'},\n",
       "    {'authorId': '50752276',\n",
       "     'name': 'Minoru Narita',\n",
       "     'url': 'https://www.semanticscholar.org/author/50752276'}],\n",
       "   'doi': '10.1016/j.jcpa.2013.08.004',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '5715c70fb6232b2b59acfc05ffc4c49212bd0a95',\n",
       "   'title': 'Experimental teschovirus encephalomyelitis in gnotobiotic pigs.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/5715c70fb6232b2b59acfc05ffc4c49212bd0a95',\n",
       "   'venue': 'Journal of comparative pathology',\n",
       "   'year': 2014},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '8154468',\n",
       "     'name': 'Nguyen Van Dung',\n",
       "     'url': 'https://www.semanticscholar.org/author/8154468'},\n",
       "    {'authorId': '119830091',\n",
       "     'name': 'Pham Hong Anh',\n",
       "     'url': 'https://www.semanticscholar.org/author/119830091'},\n",
       "    {'authorId': '7684057',\n",
       "     'name': 'Nguyen Van Cuong',\n",
       "     'url': 'https://www.semanticscholar.org/author/7684057'},\n",
       "    {'authorId': '80157047',\n",
       "     'name': 'Ng Th Hoa',\n",
       "     'url': 'https://www.semanticscholar.org/author/80157047'},\n",
       "    {'authorId': '1397905672',\n",
       "     'name': 'Juan J Carrique-Mas',\n",
       "     'url': 'https://www.semanticscholar.org/author/1397905672'},\n",
       "    {'authorId': '4650738',\n",
       "     'name': 'Vo Be Hien',\n",
       "     'url': 'https://www.semanticscholar.org/author/4650738'},\n",
       "    {'authorId': '95528271',\n",
       "     'name': 'Cassidy Sharp',\n",
       "     'url': 'https://www.semanticscholar.org/author/95528271'},\n",
       "    {'authorId': '122973135',\n",
       "     'name': 'M A Rabaa',\n",
       "     'url': 'https://www.semanticscholar.org/author/122973135'},\n",
       "    {'authorId': '122327349',\n",
       "     'name': 'A Berto',\n",
       "     'url': 'https://www.semanticscholar.org/author/122327349'},\n",
       "    {'authorId': '145576519',\n",
       "     'name': 'James I. Campbell',\n",
       "     'url': 'https://www.semanticscholar.org/author/145576519'},\n",
       "    {'authorId': '144134912',\n",
       "     'name': 'Stephen Baker',\n",
       "     'url': 'https://www.semanticscholar.org/author/144134912'},\n",
       "    {'authorId': '4054703',\n",
       "     'name': 'Jeremy J Farrar',\n",
       "     'url': 'https://www.semanticscholar.org/author/4054703'},\n",
       "    {'authorId': '2030609',\n",
       "     'name': 'M E J Woolhouse',\n",
       "     'url': 'https://www.semanticscholar.org/author/2030609'},\n",
       "    {'authorId': '2417536',\n",
       "     'name': 'Juliet E. Bryant',\n",
       "     'url': 'https://www.semanticscholar.org/author/2417536'},\n",
       "    {'authorId': '152174119',\n",
       "     'name': 'Peter Simmonds',\n",
       "     'url': 'https://www.semanticscholar.org/author/152174119'}],\n",
       "   'doi': '10.1099/jgv.0.000366',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '6f26d7d0e68218584d192daceab1334c285de089',\n",
       "   'title': 'Large-scale screening and characterization of enteroviruses and kobuviruses infecting pigs in Vietnam.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/6f26d7d0e68218584d192daceab1334c285de089',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2016},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '153215619',\n",
       "     'name': 'Monica Weismann Seabra Salles',\n",
       "     'url': 'https://www.semanticscholar.org/author/153215619'},\n",
       "    {'authorId': '3685604',\n",
       "     'name': 'Sandra F E Scholes',\n",
       "     'url': 'https://www.semanticscholar.org/author/3685604'},\n",
       "    {'authorId': '5577137',\n",
       "     'name': 'Malte Dauber',\n",
       "     'url': 'https://www.semanticscholar.org/author/5577137'},\n",
       "    {'authorId': '4925646',\n",
       "     'name': 'Gnter Strebelow',\n",
       "     'url': 'https://www.semanticscholar.org/author/4925646'},\n",
       "    {'authorId': '5106244',\n",
       "     'name': 'Chris Wojnarowicz',\n",
       "     'url': 'https://www.semanticscholar.org/author/5106244'},\n",
       "    {'authorId': '47741280',\n",
       "     'name': 'Lori Hassard',\n",
       "     'url': 'https://www.semanticscholar.org/author/47741280'},\n",
       "    {'authorId': '11706179',\n",
       "     'name': 'Andy Acton',\n",
       "     'url': 'https://www.semanticscholar.org/author/11706179'},\n",
       "    {'authorId': '5981566',\n",
       "     'name': 'Trent K. Bollinger',\n",
       "     'url': 'https://www.semanticscholar.org/author/5981566'}],\n",
       "   'doi': '10.1177/104063871102300231',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'd82e2cb8c0b7cfb2ac1a56d6bc0794caf7f50e9a',\n",
       "   'title': 'Porcine Teschovirus Polioencephalomyelitis in Western Canada',\n",
       "   'url': 'https://www.semanticscholar.org/paper/d82e2cb8c0b7cfb2ac1a56d6bc0794caf7f50e9a',\n",
       "   'venue': 'Journal of veterinary diagnostic investigation : official publication of the American Association of Veterinary Laboratory Diagnosticians, Inc',\n",
       "   'year': 2011},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '1399088265',\n",
       "     'name': 'Ndia Conceio-Neto',\n",
       "     'url': 'https://www.semanticscholar.org/author/1399088265'},\n",
       "    {'authorId': '5409759',\n",
       "     'name': 'Sebastiaan Theuns',\n",
       "     'url': 'https://www.semanticscholar.org/author/5409759'},\n",
       "    {'authorId': '49446512',\n",
       "     'name': 'Tingting Cui',\n",
       "     'url': 'https://www.semanticscholar.org/author/49446512'},\n",
       "    {'authorId': '32019123',\n",
       "     'name': 'Mark Zeller',\n",
       "     'url': 'https://www.semanticscholar.org/author/32019123'},\n",
       "    {'authorId': '6320326',\n",
       "     'name': 'Claude Kwe Yinda',\n",
       "     'url': 'https://www.semanticscholar.org/author/6320326'},\n",
       "    {'authorId': '36997712',\n",
       "     'name': 'Isaura Christiaens',\n",
       "     'url': 'https://www.semanticscholar.org/author/36997712'},\n",
       "    {'authorId': '87035089',\n",
       "     'name': 'Elisabeth Heylen',\n",
       "     'url': 'https://www.semanticscholar.org/author/87035089'},\n",
       "    {'authorId': '145734315',\n",
       "     'name': 'Marc van Ranst',\n",
       "     'url': 'https://www.semanticscholar.org/author/145734315'},\n",
       "    {'authorId': '2444772',\n",
       "     'name': 'Sebastien Christian Carpentier',\n",
       "     'url': 'https://www.semanticscholar.org/author/2444772'},\n",
       "    {'authorId': '6750977',\n",
       "     'name': 'Hans J. Nauwynck',\n",
       "     'url': 'https://www.semanticscholar.org/author/6750977'},\n",
       "    {'authorId': '145734314',\n",
       "     'name': 'Jelle Matthijnssens',\n",
       "     'url': 'https://www.semanticscholar.org/author/145734314'}],\n",
       "   'doi': '10.1093/ve/vex024',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '5e689ff66ce1f5027d5358863f244abdb996ed9b',\n",
       "   'title': 'Identification of an enterovirus recombinant with a torovirus-like gene insertion during a diarrhea outbreak in fattening pigs',\n",
       "   'url': 'https://www.semanticscholar.org/paper/5e689ff66ce1f5027d5358863f244abdb996ed9b',\n",
       "   'venue': 'Virus evolution',\n",
       "   'year': 2017},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '34229035',\n",
       "     'name': 'Srivishnupriya Anbalagan',\n",
       "     'url': 'https://www.semanticscholar.org/author/34229035'},\n",
       "    {'authorId': '145064486',\n",
       "     'name': 'Richard A. Hesse',\n",
       "     'url': 'https://www.semanticscholar.org/author/145064486'},\n",
       "    {'authorId': '3411720',\n",
       "     'name': 'Ben M. Hause',\n",
       "     'url': 'https://www.semanticscholar.org/author/3411720'}],\n",
       "   'doi': '10.1371/journal.pone.0097517',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '9ea0b70610b4afa64342511ee5ac6a3a03b28681',\n",
       "   'title': 'First Identification and Characterization of Porcine Enterovirus G in the United States',\n",
       "   'url': 'https://www.semanticscholar.org/paper/9ea0b70610b4afa64342511ee5ac6a3a03b28681',\n",
       "   'venue': 'PloS one',\n",
       "   'year': 2014},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '47446692',\n",
       "     'name': 'Mingyue Wang',\n",
       "     'url': 'https://www.semanticscholar.org/author/47446692'},\n",
       "    {'authorId': '35726280',\n",
       "     'name': 'Jia He',\n",
       "     'url': 'https://www.semanticscholar.org/author/35726280'},\n",
       "    {'authorId': '2355969',\n",
       "     'name': 'Haibing Lu',\n",
       "     'url': 'https://www.semanticscholar.org/author/2355969'},\n",
       "    {'authorId': '46398436',\n",
       "     'name': 'Yajing Liu',\n",
       "     'url': 'https://www.semanticscholar.org/author/46398436'},\n",
       "    {'authorId': '10028155',\n",
       "     'name': 'Yingrui Deng',\n",
       "     'url': 'https://www.semanticscholar.org/author/10028155'},\n",
       "    {'authorId': '6415064',\n",
       "     'name': 'Lisai Zhu',\n",
       "     'url': 'https://www.semanticscholar.org/author/6415064'},\n",
       "    {'authorId': '3867057',\n",
       "     'name': 'Changming Guo',\n",
       "     'url': 'https://www.semanticscholar.org/author/3867057'},\n",
       "    {'authorId': '1952782',\n",
       "     'name': 'Changchun Tu',\n",
       "     'url': 'https://www.semanticscholar.org/author/1952782'},\n",
       "    {'authorId': '48630475',\n",
       "     'name': 'Xinping Wang',\n",
       "     'url': 'https://www.semanticscholar.org/author/48630475'}],\n",
       "   'doi': '10.1371/journal.pone.0174600',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '669716d5da35427da4f605d643c3172e7ccfcc2e',\n",
       "   'title': 'A novel enterovirus species identified from severe diarrheal goats',\n",
       "   'url': 'https://www.semanticscholar.org/paper/669716d5da35427da4f605d643c3172e7ccfcc2e',\n",
       "   'venue': 'PloS one',\n",
       "   'year': 2017},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '2243078',\n",
       "     'name': 'Joseph Felsenstein',\n",
       "     'url': 'https://www.semanticscholar.org/author/2243078'}],\n",
       "   'doi': '10.1111/j.1558-5646.1985.tb00420.x',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '257697c50105f0681495c625e15b75e3b95d193c',\n",
       "   'title': 'CONFIDENCE LIMITS ON PHYLOGENIES: AN APPROACH USING THE BOOTSTRAP.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/257697c50105f0681495c625e15b75e3b95d193c',\n",
       "   'venue': 'Evolution; international journal of organic evolution',\n",
       "   'year': 1985},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5409759',\n",
       "     'name': 'Sebastiaan Theuns',\n",
       "     'url': 'https://www.semanticscholar.org/author/5409759'},\n",
       "    {'authorId': '1399088265',\n",
       "     'name': 'Ndia Conceio-Neto',\n",
       "     'url': 'https://www.semanticscholar.org/author/1399088265'},\n",
       "    {'authorId': '48648555',\n",
       "     'name': 'Mark Zeller',\n",
       "     'url': 'https://www.semanticscholar.org/author/48648555'},\n",
       "    {'authorId': '4644203',\n",
       "     'name': 'Elisabeth Heylen',\n",
       "     'url': 'https://www.semanticscholar.org/author/4644203'},\n",
       "    {'authorId': '3729734',\n",
       "     'name': 'Inge D. M. Roukaerts',\n",
       "     'url': 'https://www.semanticscholar.org/author/3729734'},\n",
       "    {'authorId': '145511700',\n",
       "     'name': 'Lowiese M. B. Desmarets',\n",
       "     'url': 'https://www.semanticscholar.org/author/145511700'},\n",
       "    {'authorId': '145734315',\n",
       "     'name': 'Marc van Ranst',\n",
       "     'url': 'https://www.semanticscholar.org/author/145734315'},\n",
       "    {'authorId': '6750977',\n",
       "     'name': 'Hans J. Nauwynck',\n",
       "     'url': 'https://www.semanticscholar.org/author/6750977'},\n",
       "    {'authorId': '145734314',\n",
       "     'name': 'Jelle Matthijnssens',\n",
       "     'url': 'https://www.semanticscholar.org/author/145734314'}],\n",
       "   'doi': '10.1016/j.meegid.2016.05.018',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '3ee5dd71bdc148f5c4fc4be243b056b27d04230b',\n",
       "   'title': 'Characterization of a genetically heterogeneous porcine rotavirus C, and other viruses present in the fecal virome of a non-diarrheic Belgian piglet.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/3ee5dd71bdc148f5c4fc4be243b056b27d04230b',\n",
       "   'venue': 'Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases',\n",
       "   'year': 2016},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '144032324',\n",
       "     'name': 'Andi Krumbholz',\n",
       "     'url': 'https://www.semanticscholar.org/author/144032324'},\n",
       "    {'authorId': '5577137',\n",
       "     'name': 'Malte Dauber',\n",
       "     'url': 'https://www.semanticscholar.org/author/5577137'},\n",
       "    {'authorId': '12374739',\n",
       "     'name': 'Andreas Henke',\n",
       "     'url': 'https://www.semanticscholar.org/author/12374739'},\n",
       "    {'authorId': '1401183624',\n",
       "     'name': 'Eckhard Birch-Hirschfeld',\n",
       "     'url': 'https://www.semanticscholar.org/author/1401183624'},\n",
       "    {'authorId': '5919726',\n",
       "     'name': 'Nick J. Knowles',\n",
       "     'url': 'https://www.semanticscholar.org/author/5919726'},\n",
       "    {'authorId': '8048966',\n",
       "     'name': 'Axel Stelzner',\n",
       "     'url': 'https://www.semanticscholar.org/author/8048966'},\n",
       "    {'authorId': '5898272',\n",
       "     'name': 'Roland Zell',\n",
       "     'url': 'https://www.semanticscholar.org/author/5898272'}],\n",
       "   'doi': '10.1128/JVI.76.11.5813-5821.2002',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'd4c6148254b3a0966555ce0953f5e6dd6c9f76a9',\n",
       "   'title': 'Sequencing of porcine enterovirus groups II and III reveals unique features of both virus groups.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/d4c6148254b3a0966555ce0953f5e6dd6c9f76a9',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2002},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '2993964',\n",
       "     'name': 'Makoto Nagai',\n",
       "     'url': 'https://www.semanticscholar.org/author/2993964'},\n",
       "    {'authorId': '4751067',\n",
       "     'name': 'Tsutomu Omatsu',\n",
       "     'url': 'https://www.semanticscholar.org/author/4751067'},\n",
       "    {'authorId': '34762043',\n",
       "     'name': 'Hiroshi Aoki',\n",
       "     'url': 'https://www.semanticscholar.org/author/34762043'},\n",
       "    {'authorId': '5687215',\n",
       "     'name': 'Konosuke Otomaru',\n",
       "     'url': 'https://www.semanticscholar.org/author/5687215'},\n",
       "    {'authorId': '4434509',\n",
       "     'name': 'Takehiko Uto',\n",
       "     'url': 'https://www.semanticscholar.org/author/4434509'},\n",
       "    {'authorId': '47016471',\n",
       "     'name': 'Motoya Koizumi',\n",
       "     'url': 'https://www.semanticscholar.org/author/47016471'},\n",
       "    {'authorId': '1422513581',\n",
       "     'name': 'Fujiko Minami-Fukuda',\n",
       "     'url': 'https://www.semanticscholar.org/author/1422513581'},\n",
       "    {'authorId': '4990412',\n",
       "     'name': 'Hikaru Takai',\n",
       "     'url': 'https://www.semanticscholar.org/author/4990412'},\n",
       "    {'authorId': '31448265',\n",
       "     'name': 'Toshiaki Murakami',\n",
       "     'url': 'https://www.semanticscholar.org/author/31448265'},\n",
       "    {'authorId': '4002097',\n",
       "     'name': 'Tsuneyuki Masuda',\n",
       "     'url': 'https://www.semanticscholar.org/author/4002097'},\n",
       "    {'authorId': '5656865',\n",
       "     'name': 'Hiroshi Yamasato',\n",
       "     'url': 'https://www.semanticscholar.org/author/5656865'},\n",
       "    {'authorId': '49286754',\n",
       "     'name': 'Mai Shiokawa',\n",
       "     'url': 'https://www.semanticscholar.org/author/49286754'},\n",
       "    {'authorId': '5726175',\n",
       "     'name': 'Shinobu Tsuchiaka',\n",
       "     'url': 'https://www.semanticscholar.org/author/5726175'},\n",
       "    {'authorId': '6258941',\n",
       "     'name': 'Yuki Naoi',\n",
       "     'url': 'https://www.semanticscholar.org/author/6258941'},\n",
       "    {'authorId': '6934164',\n",
       "     'name': 'Kaori Sano',\n",
       "     'url': 'https://www.semanticscholar.org/author/6934164'},\n",
       "    {'authorId': '4065992',\n",
       "     'name': 'Sachiko Okazaki',\n",
       "     'url': 'https://www.semanticscholar.org/author/4065992'},\n",
       "    {'authorId': '36375266',\n",
       "     'name': 'Yukie Katayama',\n",
       "     'url': 'https://www.semanticscholar.org/author/36375266'},\n",
       "    {'authorId': '5144887',\n",
       "     'name': 'Mami Oba',\n",
       "     'url': 'https://www.semanticscholar.org/author/5144887'},\n",
       "    {'authorId': '6199928',\n",
       "     'name': 'Tetsuya Furuya',\n",
       "     'url': 'https://www.semanticscholar.org/author/6199928'},\n",
       "    {'authorId': '4511770',\n",
       "     'name': 'Junsuke Shirai',\n",
       "     'url': 'https://www.semanticscholar.org/author/4511770'},\n",
       "    {'authorId': '49359835',\n",
       "     'name': 'Tetsuya Mizutani',\n",
       "     'url': 'https://www.semanticscholar.org/author/49359835'}],\n",
       "   'doi': '10.1007/s00705-015-2543-7',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '7b44740e1b8d1a8269f2945bbe7fdedec940deb4',\n",
       "   'title': 'Full genome analysis of bovine astrovirus from fecal samples of cattle in Japan: identification of possible interspecies transmission of bovine astrovirus',\n",
       "   'url': 'https://www.semanticscholar.org/paper/7b44740e1b8d1a8269f2945bbe7fdedec940deb4',\n",
       "   'venue': 'Archives of Virology',\n",
       "   'year': 2015},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '144215561',\n",
       "     'name': 'kos Boros',\n",
       "     'url': 'https://www.semanticscholar.org/author/144215561'},\n",
       "    {'authorId': '5601977',\n",
       "     'name': 'Pter Pankovics',\n",
       "     'url': 'https://www.semanticscholar.org/author/5601977'},\n",
       "    {'authorId': '8964093',\n",
       "     'name': 'Gbor Reuter',\n",
       "     'url': 'https://www.semanticscholar.org/author/8964093'}],\n",
       "   'doi': '10.1016/j.meegid.2011.04.003',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'f9c32878550b8678ff076ddd439de76dbe03d7c9',\n",
       "   'title': 'Characterization of a novel porcine enterovirus in domestic pig in Hungary.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/f9c32878550b8678ff076ddd439de76dbe03d7c9',\n",
       "   'venue': 'Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases',\n",
       "   'year': 2011},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6577336',\n",
       "     'name': 'Alexander N Lukashev',\n",
       "     'url': 'https://www.semanticscholar.org/author/6577336'},\n",
       "    {'authorId': '30517645',\n",
       "     'name': 'Olga E. Ivanova',\n",
       "     'url': 'https://www.semanticscholar.org/author/30517645'},\n",
       "    {'authorId': '4346185',\n",
       "     'name': 'Tatyana P Eremeeva',\n",
       "     'url': 'https://www.semanticscholar.org/author/4346185'},\n",
       "    {'authorId': '31599028',\n",
       "     'name': 'Larisa V Gmyl',\n",
       "     'url': 'https://www.semanticscholar.org/author/31599028'}],\n",
       "   'doi': '10.1128/JCM.02386-06',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'f726810ec0f069767103f71a80fef2a0c4b03df6',\n",
       "   'title': 'Analysis of echovirus 30 isolates from Russia and new independent states revealing frequent recombination and reemergence of ancient lineages.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/f726810ec0f069767103f71a80fef2a0c4b03df6',\n",
       "   'venue': 'Journal of clinical microbiology',\n",
       "   'year': 2008},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6577336',\n",
       "     'name': 'Alexander N Lukashev',\n",
       "     'url': 'https://www.semanticscholar.org/author/6577336'}],\n",
       "   'doi': '10.1002/rmv.457',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '113b4b01bf2ee79b5b4c33606b238655249d7d97',\n",
       "   'title': 'Role of recombination in evolution of enteroviruses.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/113b4b01bf2ee79b5b4c33606b238655249d7d97',\n",
       "   'venue': 'Reviews in medical virology',\n",
       "   'year': 2005},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6577336',\n",
       "     'name': 'Alexander N Lukashev',\n",
       "     'url': 'https://www.semanticscholar.org/author/6577336'},\n",
       "    {'authorId': '6095466',\n",
       "     'name': 'Vasilii A Lashkevich',\n",
       "     'url': 'https://www.semanticscholar.org/author/6095466'},\n",
       "    {'authorId': '30517645',\n",
       "     'name': 'Olga E. Ivanova',\n",
       "     'url': 'https://www.semanticscholar.org/author/30517645'},\n",
       "    {'authorId': '5072833',\n",
       "     'name': 'Galina A. Koroleva',\n",
       "     'url': 'https://www.semanticscholar.org/author/5072833'},\n",
       "    {'authorId': '2271839',\n",
       "     'name': 'A. E. Hinkkanen',\n",
       "     'url': 'https://www.semanticscholar.org/author/2271839'},\n",
       "    {'authorId': '2884366',\n",
       "     'name': 'Jorma Ilonen',\n",
       "     'url': 'https://www.semanticscholar.org/author/2884366'}],\n",
       "   'doi': '10.1099/vir.0.81264-0',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '00ac0e08d5ae1fef25383d15c70a72321acd3451',\n",
       "   'title': 'Recombination in circulating Human enterovirus B: independent evolution of structural and non-structural genome regions.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/00ac0e08d5ae1fef25383d15c70a72321acd3451',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2005},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '47462720',\n",
       "     'name': 'Jonathan Auerbach',\n",
       "     'url': 'https://www.semanticscholar.org/author/47462720'},\n",
       "    {'authorId': '47610630',\n",
       "     'name': 'Diane Prager',\n",
       "     'url': 'https://www.semanticscholar.org/author/47610630'},\n",
       "    {'authorId': '97155699',\n",
       "     'name': 'Silvia Neuhaus',\n",
       "     'url': 'https://www.semanticscholar.org/author/97155699'},\n",
       "    {'authorId': '13998343',\n",
       "     'name': 'Ulrich Loss',\n",
       "     'url': 'https://www.semanticscholar.org/author/13998343'},\n",
       "    {'authorId': '48386536',\n",
       "     'name': 'K. H. Witte',\n",
       "     'url': 'https://www.semanticscholar.org/author/48386536'}],\n",
       "   'doi': '10.1111/j.1439-0450.1994.tb00228.x',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'a06577c92a525ab957650d7811cbac2f486b008e',\n",
       "   'title': 'Grouping of porcine enteroviruses by indirect immunofluorescence and description of two new serotypes.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/a06577c92a525ab957650d7811cbac2f486b008e',\n",
       "   'venue': 'Zentralblatt fur Veterinarmedizin. Reihe B. Journal of veterinary medicine. Series B',\n",
       "   'year': 1994},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '1422912456',\n",
       "     'name': 'Etienne Simon-Loriere',\n",
       "     'url': 'https://www.semanticscholar.org/author/1422912456'},\n",
       "    {'authorId': '7006609',\n",
       "     'name': 'Edward C. Holmes',\n",
       "     'url': 'https://www.semanticscholar.org/author/7006609'}],\n",
       "   'doi': '10.1038/nrmicro2614',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'a81761e73c3fa74bdeb867c9fadf6835496cae65',\n",
       "   'title': 'Why do RNA viruses recombine?',\n",
       "   'url': 'https://www.semanticscholar.org/paper/a81761e73c3fa74bdeb867c9fadf6835496cae65',\n",
       "   'venue': 'Nature Reviews Microbiology',\n",
       "   'year': 2011},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5499074',\n",
       "     'name': 'Naina Barretto',\n",
       "     'url': 'https://www.semanticscholar.org/author/5499074'},\n",
       "    {'authorId': '1681838',\n",
       "     'name': 'Dalia Jukneliene',\n",
       "     'url': 'https://www.semanticscholar.org/author/1681838'},\n",
       "    {'authorId': '4991159',\n",
       "     'name': 'Kiira Ratia',\n",
       "     'url': 'https://www.semanticscholar.org/author/4991159'},\n",
       "    {'authorId': '48354239',\n",
       "     'name': 'Zhongbin Chen',\n",
       "     'url': 'https://www.semanticscholar.org/author/48354239'},\n",
       "    {'authorId': '5682863',\n",
       "     'name': 'Andrew D. Mesecar',\n",
       "     'url': 'https://www.semanticscholar.org/author/5682863'},\n",
       "    {'authorId': '1712809',\n",
       "     'name': 'Susan C Baker',\n",
       "     'url': 'https://www.semanticscholar.org/author/1712809'}],\n",
       "   'doi': '10.1128/JVI.79.24.15189-15198.2005',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '0c5499586765a94f41336319159d62e931ed3906',\n",
       "   'title': 'The papain-like protease of severe acute respiratory syndrome coronavirus has deubiquitinating activity.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/0c5499586765a94f41336319159d62e931ed3906',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2005},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '143782541',\n",
       "     'name': 'Peter Simmonds',\n",
       "     'url': 'https://www.semanticscholar.org/author/143782541'}],\n",
       "   'doi': '10.1128/JVI.01076-06',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '75d7201415a83c6c99fdcd423142fa65f3c8a29c',\n",
       "   'title': 'Recombination and selection in the evolution of picornaviruses and other Mammalian positive-stranded RNA viruses.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/75d7201415a83c6c99fdcd423142fa65f3c8a29c',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2006},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6577336',\n",
       "     'name': 'Alexander N Lukashev',\n",
       "     'url': 'https://www.semanticscholar.org/author/6577336'}],\n",
       "   'doi': '10.1002/rmv.660',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'a973023826a13c838fa6742863fc9ef16296eb21',\n",
       "   'title': 'Recombination among picornaviruses.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/a973023826a13c838fa6742863fc9ef16296eb21',\n",
       "   'venue': 'Reviews in medical virology',\n",
       "   'year': 2010},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '1398603304',\n",
       "     'name': 'Carita Savolainen-Kopra',\n",
       "     'url': 'https://www.semanticscholar.org/author/1398603304'},\n",
       "    {'authorId': '3684613',\n",
       "     'name': 'Soile Blomqvist',\n",
       "     'url': 'https://www.semanticscholar.org/author/3684613'}],\n",
       "   'doi': '10.1002/rmv.663',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'd19910672233d5bbc3feedf060b28ec0cdf6a096',\n",
       "   'title': 'Mechanisms of genetic variation in polioviruses.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/d19910672233d5bbc3feedf060b28ec0cdf6a096',\n",
       "   'venue': 'Reviews in medical virology',\n",
       "   'year': 2010},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6577336',\n",
       "     'name': 'Alexander N Lukashev',\n",
       "     'url': 'https://www.semanticscholar.org/author/6577336'},\n",
       "    {'authorId': '6095466',\n",
       "     'name': 'Vasilii A Lashkevich',\n",
       "     'url': 'https://www.semanticscholar.org/author/6095466'},\n",
       "    {'authorId': '5072833',\n",
       "     'name': 'Galina A. Koroleva',\n",
       "     'url': 'https://www.semanticscholar.org/author/5072833'},\n",
       "    {'authorId': '2884366',\n",
       "     'name': 'Jorma Ilonen',\n",
       "     'url': 'https://www.semanticscholar.org/author/2884366'},\n",
       "    {'authorId': '2271839',\n",
       "     'name': 'A. E. Hinkkanen',\n",
       "     'url': 'https://www.semanticscholar.org/author/2271839'}],\n",
       "   'doi': '10.1099/vir.0.19469-0',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '914b9a34943d41d5d330a759e57baac1a81e56fe',\n",
       "   'title': 'Recombination in uveitis-causing enterovirus strains.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/914b9a34943d41d5d330a759e57baac1a81e56fe',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2004},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '143662664',\n",
       "     'name': 'P H E Arruda',\n",
       "     'url': 'https://www.semanticscholar.org/author/143662664'},\n",
       "    {'authorId': '145558873',\n",
       "     'name': 'Bailey Arruda',\n",
       "     'url': 'https://www.semanticscholar.org/author/145558873'},\n",
       "    {'authorId': '5746243',\n",
       "     'name': 'Kent J. Schwartz',\n",
       "     'url': 'https://www.semanticscholar.org/author/5746243'},\n",
       "    {'authorId': '4378154',\n",
       "     'name': 'Fabio Vannucci',\n",
       "     'url': 'https://www.semanticscholar.org/author/4378154'},\n",
       "    {'authorId': '37381579',\n",
       "     'name': 'Talita Pilar Resende',\n",
       "     'url': 'https://www.semanticscholar.org/author/37381579'},\n",
       "    {'authorId': '145427259',\n",
       "     'name': 'Albert Rovira',\n",
       "     'url': 'https://www.semanticscholar.org/author/145427259'},\n",
       "    {'authorId': '15791647',\n",
       "     'name': 'Paul Sundberg',\n",
       "     'url': 'https://www.semanticscholar.org/author/15791647'},\n",
       "    {'authorId': '10100953',\n",
       "     'name': 'Jerome C. Nietfeld',\n",
       "     'url': 'https://www.semanticscholar.org/author/10100953'},\n",
       "    {'authorId': '3411720',\n",
       "     'name': 'Ben M. Hause',\n",
       "     'url': 'https://www.semanticscholar.org/author/3411720'}],\n",
       "   'doi': '10.1111/tbed.12621',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '923c771c73125cb8cecb80dd347c4f65a1bb21d2',\n",
       "   'title': 'Detection of a novel sapelovirus in central nervous tissue of pigs with polioencephalomyelitis in the USA',\n",
       "   'url': 'https://www.semanticscholar.org/paper/923c771c73125cb8cecb80dd347c4f65a1bb21d2',\n",
       "   'venue': 'Transboundary and emerging diseases',\n",
       "   'year': 2017},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '153751209',\n",
       "     'name': 'Julie Dawn Thompson',\n",
       "     'url': 'https://www.semanticscholar.org/author/153751209'},\n",
       "    {'authorId': '34569434',\n",
       "     'name': 'Toby J. Gibson',\n",
       "     'url': 'https://www.semanticscholar.org/author/34569434'},\n",
       "    {'authorId': '2016127',\n",
       "     'name': 'Frdric Plewniak',\n",
       "     'url': 'https://www.semanticscholar.org/author/2016127'},\n",
       "    {'authorId': '4170541',\n",
       "     'name': 'F Jeanmougin',\n",
       "     'url': 'https://www.semanticscholar.org/author/4170541'},\n",
       "    {'authorId': '15410046',\n",
       "     'name': 'Desmond G. Higgins',\n",
       "     'url': 'https://www.semanticscholar.org/author/15410046'}],\n",
       "   'doi': '10.1093/nar/25.24.4876',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'ad29e5310c2db455346da62508243e2c662c0ddf',\n",
       "   'title': 'The CLUSTAL_X windows interface: flexible strategies for multiple sequence alignment aided by quality analysis tools.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/ad29e5310c2db455346da62508243e2c662c0ddf',\n",
       "   'venue': 'Nucleic acids research',\n",
       "   'year': 1997},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5919726',\n",
       "     'name': 'Nick J. Knowles',\n",
       "     'url': 'https://www.semanticscholar.org/author/5919726'},\n",
       "    {'authorId': '38583081',\n",
       "     'name': 'Lucille Buckley',\n",
       "     'url': 'https://www.semanticscholar.org/author/38583081'},\n",
       "    {'authorId': '1867399',\n",
       "     'name': 'H. G. Pereira',\n",
       "     'url': 'https://www.semanticscholar.org/author/1867399'}],\n",
       "   'doi': '10.1007/BF01317552',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '203e3c150be586306dae3cdf0cc9722eaff6fdcf',\n",
       "   'title': 'Classification of porcine enteroviruses by antigenic analysis and cytopathic effects in tissue culture: Description of 3 new serotypes',\n",
       "   'url': 'https://www.semanticscholar.org/paper/203e3c150be586306dae3cdf0cc9722eaff6fdcf',\n",
       "   'venue': 'Archives of Virology',\n",
       "   'year': 2005},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '48504658',\n",
       "     'name': 'Chen Zhao',\n",
       "     'url': 'https://www.semanticscholar.org/author/48504658'},\n",
       "    {'authorId': '13505828',\n",
       "     'name': 'Mark N Collins',\n",
       "     'url': 'https://www.semanticscholar.org/author/13505828'},\n",
       "    {'authorId': '3888786',\n",
       "     'name': 'Tien-Ying Hsiang',\n",
       "     'url': 'https://www.semanticscholar.org/author/3888786'},\n",
       "    {'authorId': '80706327',\n",
       "     'name': 'Robert M. Krug',\n",
       "     'url': 'https://www.semanticscholar.org/author/80706327'}],\n",
       "   'doi': '10.1016/j.tim.2013.01.005',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'c6efbff0d5753cb658e5e489ea207c2fd9134e96',\n",
       "   'title': 'Interferon-induced ISG15 pathway: an ongoing virus-host battle.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/c6efbff0d5753cb658e5e489ea207c2fd9134e96',\n",
       "   'venue': 'Trends in microbiology',\n",
       "   'year': 2013},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '4338926',\n",
       "     'name': 'Todd P. Knutson',\n",
       "     'url': 'https://www.semanticscholar.org/author/4338926'},\n",
       "    {'authorId': '5012380',\n",
       "     'name': 'Binu T. Velayudhan',\n",
       "     'url': 'https://www.semanticscholar.org/author/5012380'},\n",
       "    {'authorId': '3274695',\n",
       "     'name': 'Douglas G Marthaler',\n",
       "     'url': 'https://www.semanticscholar.org/author/3274695'}],\n",
       "   'doi': '10.1099/jgv.0.000799',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '556af33858ec60a30b1f2d5b4f5ddaefb8849eeb',\n",
       "   'title': 'A porcine enterovirus G associated with enteric disease contains a novel papain-like cysteine protease',\n",
       "   'url': 'https://www.semanticscholar.org/paper/556af33858ec60a30b1f2d5b4f5ddaefb8849eeb',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2017},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '1399165323',\n",
       "     'name': 'Marcos Prez-Losada',\n",
       "     'url': 'https://www.semanticscholar.org/author/1399165323'},\n",
       "    {'authorId': '144250863',\n",
       "     'name': 'Miguel Arenas',\n",
       "     'url': 'https://www.semanticscholar.org/author/144250863'},\n",
       "    {'authorId': '31482056',\n",
       "     'name': 'Juan Carlos Galn',\n",
       "     'url': 'https://www.semanticscholar.org/author/31482056'},\n",
       "    {'authorId': '5114167',\n",
       "     'name': 'Ferran Palero',\n",
       "     'url': 'https://www.semanticscholar.org/author/5114167'},\n",
       "    {'authorId': '101860049',\n",
       "     'name': 'Fernando Gonzlez-Candelas',\n",
       "     'url': 'https://www.semanticscholar.org/author/101860049'}],\n",
       "   'doi': '10.1016/j.meegid.2014.12.022',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '7887669b8d05b96d5e81fbe1cfdb2e39f7c873b9',\n",
       "   'title': 'Recombination in viruses: mechanisms, methods of study, and evolutionary consequences.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/7887669b8d05b96d5e81fbe1cfdb2e39f7c873b9',\n",
       "   'venue': 'Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases',\n",
       "   'year': 2015},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '7296775',\n",
       "     'name': 'Holger A. Lindner',\n",
       "     'url': 'https://www.semanticscholar.org/author/7296775'},\n",
       "    {'authorId': '1423120725',\n",
       "     'name': 'Nasser Fotouhi-Ardakani',\n",
       "     'url': 'https://www.semanticscholar.org/author/1423120725'},\n",
       "    {'authorId': '8994275',\n",
       "     'name': 'Viktoria Lytvyn',\n",
       "     'url': 'https://www.semanticscholar.org/author/8994275'},\n",
       "    {'authorId': '145790266',\n",
       "     'name': 'Paule Lachance',\n",
       "     'url': 'https://www.semanticscholar.org/author/145790266'},\n",
       "    {'authorId': '1757865',\n",
       "     'name': 'Traian Sulea',\n",
       "     'url': 'https://www.semanticscholar.org/author/1757865'},\n",
       "    {'authorId': '39375232',\n",
       "     'name': 'Robert Mnard',\n",
       "     'url': 'https://www.semanticscholar.org/author/39375232'}],\n",
       "   'doi': '10.1128/JVI.79.24.15199-15208.2005',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '96dabe4011d201e819b9f6468cfdcdaeeb765b2e',\n",
       "   'title': 'The papain-like protease from the severe acute respiratory syndrome coronavirus is a deubiquitinating enzyme.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/96dabe4011d201e819b9f6468cfdcdaeeb765b2e',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2005},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '8154468',\n",
       "     'name': 'Nguyen Van Dung',\n",
       "     'url': 'https://www.semanticscholar.org/author/8154468'},\n",
       "    {'authorId': '119830091',\n",
       "     'name': 'Pham Hong Anh',\n",
       "     'url': 'https://www.semanticscholar.org/author/119830091'},\n",
       "    {'authorId': '7684057',\n",
       "     'name': 'Nguyen Van Cuong',\n",
       "     'url': 'https://www.semanticscholar.org/author/7684057'},\n",
       "    {'authorId': '80157047',\n",
       "     'name': 'Ng Th Hoa',\n",
       "     'url': 'https://www.semanticscholar.org/author/80157047'},\n",
       "    {'authorId': '1404219581',\n",
       "     'name': 'Juan Carrique-Mas',\n",
       "     'url': 'https://www.semanticscholar.org/author/1404219581'},\n",
       "    {'authorId': '4650738',\n",
       "     'name': 'Vo Be Hien',\n",
       "     'url': 'https://www.semanticscholar.org/author/4650738'},\n",
       "    {'authorId': '48787958',\n",
       "     'name': 'James F. Campbell',\n",
       "     'url': 'https://www.semanticscholar.org/author/48787958'},\n",
       "    {'authorId': '49770688',\n",
       "     'name': 'Stephen A. Baker',\n",
       "     'url': 'https://www.semanticscholar.org/author/49770688'},\n",
       "    {'authorId': '144048633',\n",
       "     'name': 'Jeremy J. Farrar',\n",
       "     'url': 'https://www.semanticscholar.org/author/144048633'},\n",
       "    {'authorId': '2030609',\n",
       "     'name': 'M E J Woolhouse',\n",
       "     'url': 'https://www.semanticscholar.org/author/2030609'},\n",
       "    {'authorId': '2417536',\n",
       "     'name': 'Juliet E. Bryant',\n",
       "     'url': 'https://www.semanticscholar.org/author/2417536'},\n",
       "    {'authorId': '143782541',\n",
       "     'name': 'Peter Simmonds',\n",
       "     'url': 'https://www.semanticscholar.org/author/143782541'}],\n",
       "   'doi': '10.1099/vir.0.061978-0',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'e5d5caea7d3514bebd5a511cb6bf275c6de389a3',\n",
       "   'title': 'Prevalence, genetic diversity and recombination of species G enteroviruses infecting pigs in Vietnam.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/e5d5caea7d3514bebd5a511cb6bf275c6de389a3',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2014},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '3066242',\n",
       "     'name': 'Koichiro Tamura',\n",
       "     'url': 'https://www.semanticscholar.org/author/3066242'},\n",
       "    {'authorId': '143914088',\n",
       "     'name': 'Daniel Peterson',\n",
       "     'url': 'https://www.semanticscholar.org/author/143914088'},\n",
       "    {'authorId': '48984781',\n",
       "     'name': 'Nicholas Peterson',\n",
       "     'url': 'https://www.semanticscholar.org/author/48984781'},\n",
       "    {'authorId': '48226007',\n",
       "     'name': 'Glen Stecher',\n",
       "     'url': 'https://www.semanticscholar.org/author/48226007'},\n",
       "    {'authorId': '2015366',\n",
       "     'name': 'Masatoshi Nei',\n",
       "     'url': 'https://www.semanticscholar.org/author/2015366'},\n",
       "    {'authorId': '48084476',\n",
       "     'name': 'Sudhir Kumar',\n",
       "     'url': 'https://www.semanticscholar.org/author/48084476'}],\n",
       "   'doi': '10.1093/molbev/msr121',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'c700eee5f49366deed6d15f625083709cd323485',\n",
       "   'title': 'MEGA5: molecular evolutionary genetics analysis using maximum likelihood, evolutionary distance, and maximum parsimony methods.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/c700eee5f49366deed6d15f625083709cd323485',\n",
       "   'venue': 'Molecular biology and evolution',\n",
       "   'year': 2011},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '11521742',\n",
       "     'name': 'Pengcheng Shang',\n",
       "     'url': 'https://www.semanticscholar.org/author/11521742'},\n",
       "    {'authorId': '50055658',\n",
       "     'name': 'Saurav Misra',\n",
       "     'url': 'https://www.semanticscholar.org/author/50055658'},\n",
       "    {'authorId': '3411720',\n",
       "     'name': 'Ben M. Hause',\n",
       "     'url': 'https://www.semanticscholar.org/author/3411720'},\n",
       "    {'authorId': '143866319',\n",
       "     'name': 'Ying Fang',\n",
       "     'url': 'https://www.semanticscholar.org/author/143866319'}],\n",
       "   'doi': '10.1128/JVI.00450-17',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '358668884a0233ccdf37ea2d9efc6be31659a572',\n",
       "   'title': 'A Naturally Occurring Recombinant Enterovirus Expresses a Torovirus Deubiquitinase.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/358668884a0233ccdf37ea2d9efc6be31659a572',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2017},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '48878999',\n",
       "     'name': 'Shu-Chun Chiu',\n",
       "     'url': 'https://www.semanticscholar.org/author/48878999'},\n",
       "    {'authorId': '15789453',\n",
       "     'name': 'Shu-Chia Hu',\n",
       "     'url': 'https://www.semanticscholar.org/author/15789453'},\n",
       "    {'authorId': '3308660',\n",
       "     'name': 'Chih-Cheng Chang',\n",
       "     'url': 'https://www.semanticscholar.org/author/3308660'},\n",
       "    {'authorId': '2782224',\n",
       "     'name': 'Chia-Yi Chang',\n",
       "     'url': 'https://www.semanticscholar.org/author/2782224'},\n",
       "    {'authorId': '81109026',\n",
       "     'name': 'Chin-Cheng Huang',\n",
       "     'url': 'https://www.semanticscholar.org/author/81109026'},\n",
       "    {'authorId': '4231449',\n",
       "     'name': 'Victor Pang',\n",
       "     'url': 'https://www.semanticscholar.org/author/4231449'},\n",
       "    {'authorId': '82051650',\n",
       "     'name': 'F Wang',\n",
       "     'url': 'https://www.semanticscholar.org/author/82051650'}],\n",
       "   'doi': '10.1016/j.vetmic.2012.07.031',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'a6eb6d92b5861c929f3a2f7003fb4f7a7e182965',\n",
       "   'title': 'The role of porcine teschovirus in causing diseases in endemically infected pigs.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/a6eb6d92b5861c929f3a2f7003fb4f7a7e182965',\n",
       "   'venue': 'Veterinary microbiology',\n",
       "   'year': 2012},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '145499144',\n",
       "     'name': 'Darren P. Martin',\n",
       "     'url': 'https://www.semanticscholar.org/author/145499144'},\n",
       "    {'authorId': '2942684',\n",
       "     'name': 'Ben Murrell',\n",
       "     'url': 'https://www.semanticscholar.org/author/2942684'},\n",
       "    {'authorId': '48553989',\n",
       "     'name': 'Michael David Golden',\n",
       "     'url': 'https://www.semanticscholar.org/author/48553989'},\n",
       "    {'authorId': '5860349',\n",
       "     'name': 'Arjun Khoosal',\n",
       "     'url': 'https://www.semanticscholar.org/author/5860349'},\n",
       "    {'authorId': '4495122',\n",
       "     'name': 'Brejnev Muhizi Muhire',\n",
       "     'url': 'https://www.semanticscholar.org/author/4495122'}],\n",
       "   'doi': '10.1093/ve/vev003',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'e69988c3dd169deba4583c2e8dc79f3033d1072a',\n",
       "   'title': 'RDP4: Detection and analysis of recombination patterns in virus genomes',\n",
       "   'url': 'https://www.semanticscholar.org/paper/e69988c3dd169deba4583c2e8dc79f3033d1072a',\n",
       "   'venue': 'Virus evolution',\n",
       "   'year': 2015},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6577336',\n",
       "     'name': 'Alexander N Lukashev',\n",
       "     'url': 'https://www.semanticscholar.org/author/6577336'},\n",
       "    {'authorId': '49857007',\n",
       "     'name': 'Elena Yu Shumilina',\n",
       "     'url': 'https://www.semanticscholar.org/author/49857007'},\n",
       "    {'authorId': '6452302',\n",
       "     'name': 'Ilya S. Belalov',\n",
       "     'url': 'https://www.semanticscholar.org/author/6452302'},\n",
       "    {'authorId': '30517645',\n",
       "     'name': 'Olga E. Ivanova',\n",
       "     'url': 'https://www.semanticscholar.org/author/30517645'},\n",
       "    {'authorId': '4346185',\n",
       "     'name': 'Tatyana P Eremeeva',\n",
       "     'url': 'https://www.semanticscholar.org/author/4346185'},\n",
       "    {'authorId': '46376821',\n",
       "     'name': 'Vadim Reznik',\n",
       "     'url': 'https://www.semanticscholar.org/author/46376821'},\n",
       "    {'authorId': '5244037',\n",
       "     'name': 'O E Trotsenko',\n",
       "     'url': 'https://www.semanticscholar.org/author/5244037'},\n",
       "    {'authorId': '2314970',\n",
       "     'name': 'Jan Felix Drexler',\n",
       "     'url': 'https://www.semanticscholar.org/author/2314970'},\n",
       "    {'authorId': '32170898',\n",
       "     'name': 'Christian Drosten',\n",
       "     'url': 'https://www.semanticscholar.org/author/32170898'}],\n",
       "   'doi': '10.1099/vir.0.060004-0',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '07e1cb55becc5f39e2df00780a66ac160ea52011',\n",
       "   'title': 'Recombination strategies and evolutionary dynamics of the Human enterovirus A global gene pool.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/07e1cb55becc5f39e2df00780a66ac160ea52011',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2014},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5922529',\n",
       "     'name': 'Anna M. Mielech',\n",
       "     'url': 'https://www.semanticscholar.org/author/5922529'},\n",
       "    {'authorId': '47558897',\n",
       "     'name': 'Y Chen',\n",
       "     'url': 'https://www.semanticscholar.org/author/47558897'},\n",
       "    {'authorId': '5682863',\n",
       "     'name': 'Andrew D. Mesecar',\n",
       "     'url': 'https://www.semanticscholar.org/author/5682863'},\n",
       "    {'authorId': '1712809',\n",
       "     'name': 'Susan C Baker',\n",
       "     'url': 'https://www.semanticscholar.org/author/1712809'}],\n",
       "   'doi': '10.1016/j.virusres.2014.01.025',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '3b23a2ccc6ea02347c88e5328367b72cfb6a966e',\n",
       "   'title': 'Nidovirus papain-like proteases: multifunctional enzymes with protease, deubiquitinating and deISGylating activities.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/3b23a2ccc6ea02347c88e5328367b72cfb6a966e',\n",
       "   'venue': 'Virus research',\n",
       "   'year': 2014},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5410682',\n",
       "     'name': 'Yoshiko Kaku',\n",
       "     'url': 'https://www.semanticscholar.org/author/5410682'},\n",
       "    {'authorId': '48609951',\n",
       "     'name': 'Akinori Sarai',\n",
       "     'url': 'https://www.semanticscholar.org/author/48609951'},\n",
       "    {'authorId': '47920815',\n",
       "     'name': 'Yosuke Murakami',\n",
       "     'url': 'https://www.semanticscholar.org/author/47920815'}],\n",
       "   'doi': '10.1099/0022-1317-82-2-417',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '6e507c451821972c8413233ee71b004268bee836',\n",
       "   'title': 'Genetic reclassification of porcine enteroviruses.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/6e507c451821972c8413233ee71b004268bee836',\n",
       "   'venue': 'The Journal of general virology',\n",
       "   'year': 2001},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '143782541',\n",
       "     'name': 'Peter Simmonds',\n",
       "     'url': 'https://www.semanticscholar.org/author/143782541'},\n",
       "    {'authorId': '47613167',\n",
       "     'name': 'Jon Welch',\n",
       "     'url': 'https://www.semanticscholar.org/author/47613167'}],\n",
       "   'doi': '10.1128/JVI.80.1.483-493.2006',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '6d544a484f18f6e045b2f52306a5a77fc99f8c87',\n",
       "   'title': 'Frequency and dynamics of recombination within different species of human enteroviruses.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/6d544a484f18f6e045b2f52306a5a77fc99f8c87',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2006},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5152487',\n",
       "     'name': 'Barbara Holmblat',\n",
       "     'url': 'https://www.semanticscholar.org/author/5152487'},\n",
       "    {'authorId': '6549219',\n",
       "     'name': 'Sophie Jgouic',\n",
       "     'url': 'https://www.semanticscholar.org/author/6549219'},\n",
       "    {'authorId': '6522233',\n",
       "     'name': 'Claire Muslin',\n",
       "     'url': 'https://www.semanticscholar.org/author/6522233'},\n",
       "    {'authorId': '91398729',\n",
       "     'name': 'Bruno Blondel',\n",
       "     'url': 'https://www.semanticscholar.org/author/91398729'},\n",
       "    {'authorId': '5567707',\n",
       "     'name': 'Marie-Line Joffret',\n",
       "     'url': 'https://www.semanticscholar.org/author/5567707'},\n",
       "    {'authorId': '3502724',\n",
       "     'name': 'Francis Delpeyroux',\n",
       "     'url': 'https://www.semanticscholar.org/author/3502724'}],\n",
       "   'doi': '10.1128/mBio.01119-14',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '3abefdfb2b869cfed72e25833294c62ee3eb049f',\n",
       "   'title': 'Nonhomologous Recombination between Defective Poliovirus and Coxsackievirus Genomes Suggests a New Model of Genetic Plasticity for Picornaviruses',\n",
       "   'url': 'https://www.semanticscholar.org/paper/3abefdfb2b869cfed72e25833294c62ee3eb049f',\n",
       "   'venue': 'mBio',\n",
       "   'year': 2014},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '104289179',\n",
       "     'name': 'Mar\\xada Jos Vilar',\n",
       "     'url': 'https://www.semanticscholar.org/author/104289179'},\n",
       "    {'authorId': '1770742',\n",
       "     'name': 'Bibiana Peralta',\n",
       "     'url': 'https://www.semanticscholar.org/author/1770742'},\n",
       "    {'authorId': '83175085',\n",
       "     'name': 'Ignacio Garca-Bocanegra',\n",
       "     'url': 'https://www.semanticscholar.org/author/83175085'},\n",
       "    {'authorId': '1398286985',\n",
       "     'name': 'Meritxell Simon-Grif',\n",
       "     'url': 'https://www.semanticscholar.org/author/1398286985'},\n",
       "    {'authorId': '50231881',\n",
       "     'name': 'Albert Bensaid',\n",
       "     'url': 'https://www.semanticscholar.org/author/50231881'},\n",
       "    {'authorId': '143877384',\n",
       "     'name': 'Jordi Casal',\n",
       "     'url': 'https://www.semanticscholar.org/author/143877384'},\n",
       "    {'authorId': '145281629',\n",
       "     'name': 'Joaquim Segals',\n",
       "     'url': 'https://www.semanticscholar.org/author/145281629'},\n",
       "    {'authorId': '1402210898',\n",
       "     'name': 'Sonia Pina-Pedrero',\n",
       "     'url': 'https://www.semanticscholar.org/author/1402210898'}],\n",
       "   'doi': '10.1016/j.virusres.2016.01.019',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '0fcd99f1c37ff31bd642c483bfc299c26715ae65',\n",
       "   'title': 'Distribution and genetic characterization of Enterovirus G and Sapelovirus A in six Spanish swine herds.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/0fcd99f1c37ff31bd642c483bfc299c26715ae65',\n",
       "   'venue': 'Virus research',\n",
       "   'year': 2016},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5248266',\n",
       "     'name': 'Wencheng Lin',\n",
       "     'url': 'https://www.semanticscholar.org/author/5248266'},\n",
       "    {'authorId': '52209146',\n",
       "     'name': 'Shangjin Cui',\n",
       "     'url': 'https://www.semanticscholar.org/author/52209146'},\n",
       "    {'authorId': '5898272',\n",
       "     'name': 'Roland Zell',\n",
       "     'url': 'https://www.semanticscholar.org/author/5898272'}],\n",
       "   'doi': '10.1007/s00705-012-1315-x',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '3d0ccfeb96ec3e982ffd29e77d6065a7ac346ae2',\n",
       "   'title': 'Phylogeny and evolution of porcine teschovirus 8 isolated from pigs in China with reproductive failure',\n",
       "   'url': 'https://www.semanticscholar.org/paper/3d0ccfeb96ec3e982ffd29e77d6065a7ac346ae2',\n",
       "   'venue': 'Archives of Virology',\n",
       "   'year': 2012},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '5898272',\n",
       "     'name': 'Roland Zell',\n",
       "     'url': 'https://www.semanticscholar.org/author/5898272'},\n",
       "    {'authorId': '9955238',\n",
       "     'name': 'Mario Dauber',\n",
       "     'url': 'https://www.semanticscholar.org/author/9955238'},\n",
       "    {'authorId': '144032324',\n",
       "     'name': 'Andi Krumbholz',\n",
       "     'url': 'https://www.semanticscholar.org/author/144032324'},\n",
       "    {'authorId': '32901516',\n",
       "     'name': 'Andreas Henke',\n",
       "     'url': 'https://www.semanticscholar.org/author/32901516'},\n",
       "    {'authorId': '1401183624',\n",
       "     'name': 'Eckhard Birch-Hirschfeld',\n",
       "     'url': 'https://www.semanticscholar.org/author/1401183624'},\n",
       "    {'authorId': '48287389',\n",
       "     'name': 'Axel Stelzner',\n",
       "     'url': 'https://www.semanticscholar.org/author/48287389'},\n",
       "    {'authorId': '47610630',\n",
       "     'name': 'Diane Prager',\n",
       "     'url': 'https://www.semanticscholar.org/author/47610630'},\n",
       "    {'authorId': '47229508',\n",
       "     'name': 'Rdiger Wurm',\n",
       "     'url': 'https://www.semanticscholar.org/author/47229508'}],\n",
       "   'doi': '10.1128/JVI.75.4.1620-1631.2001',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'f7c15736c10d0b939f2ae87e437f60844dc1d3f2',\n",
       "   'title': 'Porcine teschoviruses comprise at least eleven distinct serotypes: molecular and evolutionary aspects.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/f7c15736c10d0b939f2ae87e437f60844dc1d3f2',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2001},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6806429',\n",
       "     'name': 'Jana Prodlalov',\n",
       "     'url': 'https://www.semanticscholar.org/author/6806429'}],\n",
       "   'doi': '10.1016/j.meegid.2012.04.025',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'c5d17a3fd0650edd930f2bd051b3af6c28ae362f',\n",
       "   'title': 'The survey of porcine teschoviruses, sapeloviruses and enteroviruses B infecting domestic pigs and wild boars in the Czech Republic between 2005 and 2011.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/c5d17a3fd0650edd930f2bd051b3af6c28ae362f',\n",
       "   'venue': 'Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases',\n",
       "   'year': 2012},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '6577336',\n",
       "     'name': 'Alexander N Lukashev',\n",
       "     'url': 'https://www.semanticscholar.org/author/6577336'},\n",
       "    {'authorId': '6095466',\n",
       "     'name': 'Vasilii A Lashkevich',\n",
       "     'url': 'https://www.semanticscholar.org/author/6095466'},\n",
       "    {'authorId': '30517645',\n",
       "     'name': 'Olga E. Ivanova',\n",
       "     'url': 'https://www.semanticscholar.org/author/30517645'},\n",
       "    {'authorId': '5072833',\n",
       "     'name': 'Galina A. Koroleva',\n",
       "     'url': 'https://www.semanticscholar.org/author/5072833'},\n",
       "    {'authorId': '2271839',\n",
       "     'name': 'A. E. Hinkkanen',\n",
       "     'url': 'https://www.semanticscholar.org/author/2271839'},\n",
       "    {'authorId': '2884366',\n",
       "     'name': 'Jorma Ilonen',\n",
       "     'url': 'https://www.semanticscholar.org/author/2884366'}],\n",
       "   'doi': '10.1128/JVI.77.19.10423-10431.2003',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': '267edccb3d3ffd4becd8b449d892f5bd8a60a2d5',\n",
       "   'title': 'Recombination in circulating enteroviruses.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/267edccb3d3ffd4becd8b449d892f5bd8a60a2d5',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2003},\n",
       "  {'arxivId': None,\n",
       "   'authors': [{'authorId': '50549640',\n",
       "     'name': 'Wei Zhang',\n",
       "     'url': 'https://www.semanticscholar.org/author/50549640'},\n",
       "    {'authorId': '6425126',\n",
       "     'name': 'Shixing Yang',\n",
       "     'url': 'https://www.semanticscholar.org/author/6425126'},\n",
       "    {'authorId': '46417311',\n",
       "     'name': 'Quan Shen',\n",
       "     'url': 'https://www.semanticscholar.org/author/46417311'},\n",
       "    {'authorId': '49381022',\n",
       "     'name': 'Liping Ren',\n",
       "     'url': 'https://www.semanticscholar.org/author/49381022'},\n",
       "    {'authorId': '6598396',\n",
       "     'name': 'Tongling Shan',\n",
       "     'url': 'https://www.semanticscholar.org/author/6598396'},\n",
       "    {'authorId': '6012799',\n",
       "     'name': 'Jianzhong Wei',\n",
       "     'url': 'https://www.semanticscholar.org/author/6012799'},\n",
       "    {'authorId': '152439604',\n",
       "     'name': 'Li Juan Cui',\n",
       "     'url': 'https://www.semanticscholar.org/author/152439604'},\n",
       "    {'authorId': '144672285',\n",
       "     'name': 'Xiuguo Hua',\n",
       "     'url': 'https://www.semanticscholar.org/author/144672285'}],\n",
       "   'doi': '10.1128/JVI.00711-12',\n",
       "   'intent': [],\n",
       "   'isInfluential': False,\n",
       "   'paperId': 'bbb3cc3b0a02ca80a8d2a54e911663a0a77bec02',\n",
       "   'title': 'Complete genome sequence of a novel porcine enterovirus strain in China.',\n",
       "   'url': 'https://www.semanticscholar.org/paper/bbb3cc3b0a02ca80a8d2a54e911663a0a77bec02',\n",
       "   'venue': 'Journal of virology',\n",
       "   'year': 2012}],\n",
       " 'title': 'Genetic diversity and recombination of enterovirus G strains in Japanese pigs: High prevalence of strains carrying a papain-like cysteine protease sequence in the enterovirus G population',\n",
       " 'topics': [{'topic': 'Papain',\n",
       "   'topicId': '35743',\n",
       "   'url': 'https://www.semanticscholar.org/topic/35743'},\n",
       "  {'topic': 'Enterovirus',\n",
       "   'topicId': '2381',\n",
       "   'url': 'https://www.semanticscholar.org/topic/2381'},\n",
       "  {'topic': 'Nucleotides',\n",
       "   'topicId': '1223',\n",
       "   'url': 'https://www.semanticscholar.org/topic/1223'},\n",
       "  {'topic': 'Endopeptidases',\n",
       "   'topicId': '2819',\n",
       "   'url': 'https://www.semanticscholar.org/topic/2819'},\n",
       "  {'topic': 'Calpain',\n",
       "   'topicId': '30396',\n",
       "   'url': 'https://www.semanticscholar.org/topic/30396'},\n",
       "  {'topic': 'Variation (Genetics)',\n",
       "   'topicId': '2512',\n",
       "   'url': 'https://www.semanticscholar.org/topic/2512'},\n",
       "  {'topic': 'Diarrhea',\n",
       "   'topicId': '1071',\n",
       "   'url': 'https://www.semanticscholar.org/topic/1071'},\n",
       "  {'topic': 'Classification',\n",
       "   'topicId': '7012',\n",
       "   'url': 'https://www.semanticscholar.org/topic/7012'},\n",
       "  {'topic': 'Cerebral Palsy',\n",
       "   'topicId': '426',\n",
       "   'url': 'https://www.semanticscholar.org/topic/426'},\n",
       "  {'topic': 'Virus',\n",
       "   'topicId': '350355',\n",
       "   'url': 'https://www.semanticscholar.org/topic/350355'},\n",
       "  {'topic': 'Hereditary Diseases',\n",
       "   'topicId': '4660',\n",
       "   'url': 'https://www.semanticscholar.org/topic/4660'},\n",
       "  {'topic': 'Metagenomics',\n",
       "   'topicId': '103996',\n",
       "   'url': 'https://www.semanticscholar.org/topic/103996'},\n",
       "  {'topic': 'Recombination, Genetic',\n",
       "   'topicId': '40489',\n",
       "   'url': 'https://www.semanticscholar.org/topic/40489'},\n",
       "  {'topic': 'Sequence Analysis',\n",
       "   'topicId': '174547',\n",
       "   'url': 'https://www.semanticscholar.org/topic/174547'},\n",
       "  {'topic': 'Epidermodysplasia Verruciformis',\n",
       "   'topicId': '247480',\n",
       "   'url': 'https://www.semanticscholar.org/topic/247480'},\n",
       "  {'topic': 'Molecular Epidemiology',\n",
       "   'topicId': '567811',\n",
       "   'url': 'https://www.semanticscholar.org/topic/567811'},\n",
       "  {'topic': 'Carrying',\n",
       "   'topicId': '1217',\n",
       "   'url': 'https://www.semanticscholar.org/topic/1217'}],\n",
       " 'url': 'https://www.semanticscholar.org/paper/eceb1667ac6722e9131eb7d0f560a7f76bfb25f1',\n",
       " 'venue': 'PloS one',\n",
       " 'year': 2018}"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "api_data[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['abstract', 'arxivId', 'authors', 'citationVelocity', 'citations', 'corpusId', 'doi', 'fieldsOfStudy', 'influentialCitationCount', 'is_open_access', 'is_publisher_licensed', 'paperId', 'references', 'title', 'topics', 'url', 'venue', 'year'])"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "api_data[0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11390"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "len(api_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# build a dataframe\n",
    "columns = ['pid','doi','title','abstract',\n",
    "           'authors', 'venue','year']\n",
    "rows = []\n",
    "for item in data:\n",
    "    # define a row\n",
    "    row = {col_name : '' for col_name in columns} #pid,doi,pmid,title,abstract,authors,pub_date,venue,year = '','','','','','','','',''\n",
    "    row['pid'] = item['paper_id']\n",
    "    if 'paper_id' in item:\n",
    "        pid = item['paper_id']\n",
    "    \n",
    "        if pid in api_dict:\n",
    "            api_item = api_dict[pid]\n",
    "            for key in columns:\n",
    "                if key in api_item and key!='authors':\n",
    "                    row[key] = str(api_item[key])\n",
    "        else:\n",
    "            \n",
    "            row['title'] = item['metadata']['title']\n",
    "            \n",
    "            abstract = item['abstract']\n",
    "            abs_text = ''\n",
    "            if type(abstract) == list and len(abstract)>0:\n",
    "                for part in abstract:\n",
    "                    if 'text' in part and len(abs_text)==0:\n",
    "                        abs_text = part['text']\n",
    "                    else:\n",
    "                        abs_text = abs_text+'. '+part['text']\n",
    "                    if len(abs_text.split())>300:\n",
    "                        break\n",
    "                        \n",
    "            elif 'body_text' in item:\n",
    "                for text_unit in item['body_text']:\n",
    "                    if 'text' in text_unit:\n",
    "                        if len(abs_text)==0:\n",
    "                            abs_text = text_unit['text']\n",
    "                        else:\n",
    "                            abs_text = abs_text +'. ' + text_unit['text']\n",
    "                    if len(abs_text.split())>300:\n",
    "                        break\n",
    "                \n",
    "            row['abstract'] = abs_text\n",
    "                \n",
    "        if 'authors' in item['metadata']:\n",
    "            row['authors'] = '; '.join([x['last']+', '+x['first'] for x in item['metadata']['authors']])\n",
    "        rows.append(row)\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13202, 7)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(rows, columns = columns)\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pid</th>\n",
       "      <th>doi</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>authors</th>\n",
       "      <th>venue</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>2704</td>\n",
       "      <td>2a7c951e191425fd9fa5ac108f07a1f02eb75872</td>\n",
       "      <td>10.1186/2047-9158-1-9</td>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "      <td>Luo, Xiao-Guang; Chen, Sheng-Di</td>\n",
       "      <td>Translational Neurodegeneration</td>\n",
       "      <td>2012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10259</td>\n",
       "      <td>fc7c462f4b3c1a7e4ab9cf9d7be29353a8595a8c</td>\n",
       "      <td>10.1128/mBio.00002-13</td>\n",
       "      <td>Human Coronavirus EMC Is Not the Same as Sever...</td>\n",
       "      <td>A newly identified betacoronavirus, human coro...</td>\n",
       "      <td>Perlman, Stanley; Zhao, Jincun</td>\n",
       "      <td>mBio</td>\n",
       "      <td>2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>748</td>\n",
       "      <td>9fb5ca87ca47d35d87161158ae37fc0ca642c51e</td>\n",
       "      <td></td>\n",
       "      <td>A Mac2-positive progenitor-like microglial pop...</td>\n",
       "      <td>10</td>\n",
       "      <td>Zhan, Lihong; Sohn, Peter; Zhou, Yungui; Li, Y...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            pid                    doi  \\\n",
       "2704   2a7c951e191425fd9fa5ac108f07a1f02eb75872  10.1186/2047-9158-1-9   \n",
       "10259  fc7c462f4b3c1a7e4ab9cf9d7be29353a8595a8c  10.1128/mBio.00002-13   \n",
       "748    9fb5ca87ca47d35d87161158ae37fc0ca642c51e                          \n",
       "\n",
       "                                                   title  \\\n",
       "2704   The changing phenotype of microglia from homeo...   \n",
       "10259  Human Coronavirus EMC Is Not the Same as Sever...   \n",
       "748    A Mac2-positive progenitor-like microglial pop...   \n",
       "\n",
       "                                                abstract  \\\n",
       "2704   It has been nearly a century since the early d...   \n",
       "10259  A newly identified betacoronavirus, human coro...   \n",
       "748                                                   10   \n",
       "\n",
       "                                                 authors  \\\n",
       "2704                     Luo, Xiao-Guang; Chen, Sheng-Di   \n",
       "10259                     Perlman, Stanley; Zhao, Jincun   \n",
       "748    Zhan, Lihong; Sohn, Peter; Zhou, Yungui; Li, Y...   \n",
       "\n",
       "                                 venue  year  \n",
       "2704   Translational Neurodegeneration  2012  \n",
       "10259                             mBio  2013  \n",
       "748                                           "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sample(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At the time of writing, some items appear to be missing from the API.  Possibly this is temporary and we will see the items showing up soon.\n",
    "\n",
    "Checks for data quality & consistency."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pid : 0 / 13202\n",
      "doi : 5921 / 13202\n",
      "title : 568 / 13202\n",
      "abstract : 0 / 13202\n",
      "authors : 861 / 13202\n",
      "venue : 6012 / 13202\n",
      "year : 5921 / 13202\n"
     ]
    }
   ],
   "source": [
    "# how much missing data is there\n",
    "for column in columns:\n",
    "    print(column,':', df[df[column]==''].shape[0], '/',df.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pid</th>\n",
       "      <th>doi</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>authors</th>\n",
       "      <th>venue</th>\n",
       "      <th>year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>11249</td>\n",
       "      <td>8cb5954efbc7ab4a0d43db196b9b58b3838945b0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Viroporins are small virus-encoded proteins th...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13025</td>\n",
       "      <td>4fc3b508543068a9d54f11f0ecf52580df768307</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>The Centers for Disease Control and Prevention...</td>\n",
       "      <td>Rao, Carol; Goryoka, Grace; Henao, Olga; Clark...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12866</td>\n",
       "      <td>1447cbaffd1c03718da0534b7bfa97435cd3c78d</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Animals at Chulalongkorn University (Bangkok, ...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            pid doi title  \\\n",
       "11249  8cb5954efbc7ab4a0d43db196b9b58b3838945b0             \n",
       "13025  4fc3b508543068a9d54f11f0ecf52580df768307             \n",
       "12866  1447cbaffd1c03718da0534b7bfa97435cd3c78d             \n",
       "\n",
       "                                                abstract  \\\n",
       "11249  Viroporins are small virus-encoded proteins th...   \n",
       "13025  The Centers for Disease Control and Prevention...   \n",
       "12866  Animals at Chulalongkorn University (Bangkok, ...   \n",
       "\n",
       "                                                 authors venue year  \n",
       "11249                                                                \n",
       "13025  Rao, Carol; Goryoka, Grace; Henao, Olga; Clark...             \n",
       "12866                                                                "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['title']==''].sample(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "tiabs = []\n",
    "for i,row in df.iterrows():\n",
    "    if len(row['title'])==0:\n",
    "        tiabs.append(row['abstract'])\n",
    "    else:\n",
    "        tiabs.append(row['title'] + '. ' + row['abstract'])\n",
    "df['tiabs'] = tiabs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('data/s2_data.csv', encoding = 'utf-8-sig', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add CrossRef data\n",
    "There are a lot of DOIs in the data. Unfortunately, many of them are not CrossRef DOIs (either DOIs from other registrars or errors). The functions I have for retrieving DOIs are only meant for getting a list of CrossRef DOIs, so they are a little slow here. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# get crossref data incase you need it\n",
    "import urllib\n",
    "import requests\n",
    "import time\n",
    "from datetime import datetime as dt\n",
    "\n",
    "\n",
    "def chunks(l, n):\n",
    "    '''\n",
    "    converts a list into a list of lists each with max length == n\n",
    "    '''\n",
    "    # For item i in a range that is a length of l,\n",
    "    for i in range(0, len(l), n):\n",
    "        # Create an index range for l of n items:\n",
    "        yield l[i:i+n]\n",
    "\n",
    "def get_one_crossref(doi, headers, email = 'adamday1284@gmail.com'):\n",
    "    url = r'http://api.crossref.org/works/{}'.format(urllib.parse.quote(doi))\n",
    "    r = requests.get(url, headers = headers)\n",
    "    try:\n",
    "        if r.json()['status']!='failed':\n",
    "            return [r.json()['message']] # output should be a list for consistency\n",
    "        else:\n",
    "            print('failure for doi:',doi)\n",
    "            print('(skipping)')\n",
    "            raise # or return None?\n",
    "    except:\n",
    "        print('Error in response.  Not interpretable as json?')\n",
    "        print('Searched for:', doi)\n",
    "        print('Trying unescaped doi')\n",
    "        url = r'http://api.crossref.org/works/{}'.format(doi)\n",
    "        r = requests.get(url, headers = headers)\n",
    "        try:\n",
    "            js = r.json()\n",
    "            if r.json()['status']!='failed':\n",
    "                return [r.json()['message']] # output should be a list for consistency\n",
    "            else:\n",
    "                print('failure for doi:',doi)\n",
    "                print('(skipping)')\n",
    "                return None\n",
    "            print('taking a break - sleep for 10s')\n",
    "            time.sleep(10)\n",
    "            return None\n",
    "        except Exception as e:\n",
    "            print('Error: ',e)\n",
    "            return None\n",
    "\n",
    "def get_many_crossref(dois, email = ''):\n",
    "    '''\n",
    "    Pulls a list of CR DOIs all at once.\n",
    "    Yields output json for each batch for analysis/caching.\n",
    "    -- Note that batch_size==100 seems to be a limit --\n",
    "    packages: urllib, requests, time\n",
    "    functions: chunks\n",
    "    '''\n",
    "\n",
    "    headers = {\n",
    "    'User-Agent': 'Data analysis',\n",
    "    'mailto': email\n",
    "    }\n",
    "\n",
    "    # make generator of lists\n",
    "    batch_size = 100\n",
    "    doi_batches = chunks(dois,batch_size)\n",
    "    i = 0\n",
    "    for doi_batch in doi_batches:\n",
    "\n",
    "        # make query string\n",
    "        dois_s = 'doi:' + ',doi:'.join(doi_batch)\n",
    "        # Crossref documentation recommends %-encoding dois\n",
    "        dois_s = urllib.parse.quote(dois_s)\n",
    "        # build query url\n",
    "        url = r'http://api.crossref.org/works/?filter={}&rows={}'.format(dois_s, batch_size)\n",
    "        # make request\n",
    "        r = requests.get(url, headers = headers)\n",
    "\n",
    "        # check response time.\n",
    "        # Documentation recommends backing off if response time is too long.\n",
    "        response_time = r.elapsed.total_seconds()\n",
    "        print(response_time, 'seconds for last request')\n",
    "\n",
    "        # responses are generally <1s.\n",
    "        # simple rule for sleeping if responses are slow\n",
    "        if response_time > 2.0:\n",
    "            print('CrossRef slow to respond. Sleep for a few seconds.')\n",
    "            time.sleep(int(response_time))\n",
    "            print('Finished sleeping')\n",
    "\n",
    "        data = r.json()\n",
    "        i+=1\n",
    "\n",
    "        if data['status'] != 'failed':\n",
    "            records = data['message']['items']\n",
    "            n_results = len(records)\n",
    "\n",
    "            print(dt.now())\n",
    "            print(i, 'requests done')\n",
    "\n",
    "            print(batch_size, 'items requested')\n",
    "            print(n_results,'items returned')\n",
    "            print('---------------------------------')\n",
    "\n",
    "            out = data['message']['items']\n",
    "\n",
    "            if n_results < batch_size:\n",
    "                dois_returned = [record['DOI'] for record in records]\n",
    "                failed_dois = [doi for doi in doi_batch\n",
    "                                if doi not in dois_returned]\n",
    "                if len(failed_dois)>0:\n",
    "                    print('Retrieving remaining DOIs one at a time')\n",
    "                    for doi in failed_dois:\n",
    "                        doi_result = get_one_crossref(doi=doi, headers=headers)\n",
    "                        if doi_result == None:\n",
    "                            pass\n",
    "                        else:\n",
    "                            if type(doi_result)==dict and 'message' in doi_result:\n",
    "                                out.append(doi_result['message'])\n",
    "                            else:\n",
    "                                out=None\n",
    "\n",
    "\n",
    "            yield out\n",
    "        else:\n",
    "            # if there's a failure, it's likely because of a typo in a doi\n",
    "            # this should retrieve all other dois in the batch\n",
    "            print('Failure! - invalid doi somewhere in batch?')\n",
    "            print('iterating over each item in batch:')\n",
    "            for doi in doi_batch:\n",
    "                out = get_one_crossref(doi=doi, headers = headers)\n",
    "                if out != None:\n",
    "                    yield out\n",
    "                else:\n",
    "                    pass\n",
    "\n",
    "\n",
    "\n",
    "# get all crossref dates\n",
    "from dateutil import parser\n",
    "import datetime\n",
    "from dateutil.tz import tzutc\n",
    "\n",
    "\"\"\"\n",
    "The next fn is a result of CrossRef not always having an accurate publication\n",
    "date.  So, we take a best-guess that the earliest date associated with the\n",
    "record is probably the publication date, or the closest thing to it.\n",
    "\"\"\"\n",
    "from dateutil import parser\n",
    "def cr_earliest_date(works_record):\n",
    "    '''\n",
    "    Takes a CrossRef works record and returns the earliest date associated with it.\n",
    "    packages: dateutil.parser\n",
    "    '''\n",
    "    date_types = ['created', 'deposited', 'indexed', 'issued',\n",
    "                  'published-online', 'published-print']\n",
    "    dates = []\n",
    "    for date_type in date_types:\n",
    "        # date_types are often missing in CrossRef works records, so we're checking\n",
    "        if date_type in works_record:\n",
    "            date_dict = works_record[date_type]\n",
    "            for thing in date_dict:\n",
    "                try:\n",
    "                    dates.append(parser.parse(date_dict[thing]))\n",
    "                except:\n",
    "                    # in case the parser fails\n",
    "                    pass\n",
    "        else:\n",
    "            pass\n",
    "    earliest = min(dates)\n",
    "    return earliest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.344814 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:24:51.892462\n",
      "1 requests done\n",
      "100 items requested\n",
      "100 items returned\n",
      "---------------------------------\n",
      "4.917929 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:25:01.649535\n",
      "2 requests done\n",
      "100 items requested\n",
      "97 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.21010/ajid.v12i1.5\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3390/ijerph121013289\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3390/v7102891\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.551685 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:25:19.252123\n",
      "3 requests done\n",
      "100 items requested\n",
      "100 items returned\n",
      "---------------------------------\n",
      "2.526133 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:25:27.601571\n",
      "4 requests done\n",
      "100 items requested\n",
      "98 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.22038/ijbms.2016.7471\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.4322/acr.%y.26080\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.243937 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:25:42.391024\n",
      "5 requests done\n",
      "100 items requested\n",
      "99 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3904/kjim.2018.036\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "3.772274 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:25:58.024836\n",
      "6 requests done\n",
      "100 items requested\n",
      "97 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.1590/S1517-838220120001000011\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.1234/1201\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.15167/2421-4248/jpmh2014.55.3.432\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.910331 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:26:12.494559\n",
      "7 requests done\n",
      "100 items requested\n",
      "99 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.30466/vrf.2018.32089\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.096811 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:26:30.746759\n",
      "8 requests done\n",
      "100 items requested\n",
      "99 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3134/ehtj.10.006\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.219827 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:26:47.245603\n",
      "9 requests done\n",
      "100 items requested\n",
      "97 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.12890/2019_001155\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.2307/3429178\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.22038/ajp.2016.6455\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "1.664195 seconds for last request\n",
      "2020-03-18 11:27:03.166839\n",
      "10 requests done\n",
      "100 items requested\n",
      "99 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3904/kjim.2018.031\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.610264 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:27:20.309935\n",
      "11 requests done\n",
      "100 items requested\n",
      "97 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3390/v10120702\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.15167/2421-4248/jpmh2014.55.4.452\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.5851/kosfa.2018.38.3.615\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "1.598407 seconds for last request\n",
      "2020-03-18 11:27:31.863026\n",
      "12 requests done\n",
      "100 items requested\n",
      "98 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.22114/ajem.v0i0.121\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.22037/emergency.v3i1.7419\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "1.563836 seconds for last request\n",
      "2020-03-18 11:27:44.108117\n",
      "13 requests done\n",
      "100 items requested\n",
      "100 items returned\n",
      "---------------------------------\n",
      "1.620961 seconds for last request\n",
      "2020-03-18 11:27:47.580112\n",
      "14 requests done\n",
      "100 items requested\n",
      "100 items returned\n",
      "---------------------------------\n",
      "1.821997 seconds for last request\n",
      "2020-03-18 11:27:54.727654\n",
      "15 requests done\n",
      "100 items requested\n",
      "97 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3201/eid1109.041344\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.13918/j.issn.2095-8137.2017.004\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3205/dgkh000326\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "1.561739 seconds for last request\n",
      "2020-03-18 11:28:05.753762\n",
      "16 requests done\n",
      "100 items requested\n",
      "99 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.1590/0074-0276108052013010\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "1.760264 seconds for last request\n",
      "2020-03-18 11:28:21.005693\n",
      "17 requests done\n",
      "100 items requested\n",
      "97 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.9204/aogh.2381\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3201/eid/1706.101381\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.5365/2019.10.1.006\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.6682 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:28:36.747424\n",
      "18 requests done\n",
      "100 items requested\n",
      "99 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.7189/johg.08.020310\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.659428 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:28:50.362018\n",
      "19 requests done\n",
      "100 items requested\n",
      "100 items returned\n",
      "---------------------------------\n",
      "2.53649 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:28:59.695486\n",
      "20 requests done\n",
      "100 items requested\n",
      "98 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.22037/ijpr.2012.1139\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.22037/ijpr.2015.1666\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.297392 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:29:14.508644\n",
      "21 requests done\n",
      "100 items requested\n",
      "98 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.1590/S1517-838220120004000049\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.14639/0392-100X-1743\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.44866 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:29:33.858830\n",
      "22 requests done\n",
      "100 items requested\n",
      "96 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.4314/ahs.v19i4.34\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3329/jhpn.v25i3.642\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.15167/2421-4248/jpmh2019.60.3.1314\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.4137/DTI.S0\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.538902 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:29:52.220322\n",
      "23 requests done\n",
      "100 items requested\n",
      "100 items returned\n",
      "---------------------------------\n",
      "1.99931 seconds for last request\n",
      "2020-03-18 11:29:57.684048\n",
      "24 requests done\n",
      "100 items requested\n",
      "99 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3390/v7102893\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "3.353988 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:30:14.216430\n",
      "25 requests done\n",
      "100 items requested\n",
      "97 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.4137/BMI.S0\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.5847/wjem.j.1920-8642.2013.02.002\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3969/j.issn.1002-0829.2012.03.001\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "2.122612 seconds for last request\n",
      "CrossRef slow to respond. Sleep for a few seconds.\n",
      "Finished sleeping\n",
      "2020-03-18 11:30:30.315994\n",
      "26 requests done\n",
      "100 items requested\n",
      "98 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.3205/dgkh000314\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "Error in response.  Not interpretable as json?\n",
      "Searched for: 10.5365/wpsar.2018.9.5.013\n",
      "Trying unescaped doi\n",
      "Error:  Expecting value: line 1 column 1 (char 0)\n",
      "0.836702 seconds for last request\n",
      "2020-03-18 11:30:40.433334\n",
      "27 requests done\n",
      "100 items requested\n",
      "15 items returned\n",
      "---------------------------------\n",
      "Retrieving remaining DOIs one at a time\n",
      "Wall time: 6min 5s\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "5600"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "cr_data_path = os.path.abspath('data/cr_data.json')\n",
    "cr_data = []\n",
    "if os.path.exists(cr_data_path):\n",
    "    with open(cr_data_path, 'r') as f:\n",
    "        cr_data = json.load(f)\n",
    "cr_dict = {x['DOI']:x for x in cr_data}  \n",
    "print('Already got data for ', len(cr_dict), ' crossref dois.')\n",
    "dois = list(set([x for x in df['doi'].tolist() if x not in {'',None,'None'} and x not in cr_dict]))\n",
    "for x in get_many_crossref(dois, email = os.environ['myemail']):\n",
    "    if type(x)==list:\n",
    "        cr_data.extend(x)\n",
    "        \n",
    "with open(cr_data_path,'w') as f:\n",
    "    json.dump(cr_data,f)\n",
    "cr_dict = {x['DOI']:x for x in cr_data}  \n",
    "len(cr_data), len(cr_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "cr_dict = {x['DOI']:x for x in cr_data}  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "cr_journals = []\n",
    "cr_journals_short = []\n",
    "cr_issns = []\n",
    "publishers = []\n",
    "pubdates = []\n",
    "for doi in df['doi'].tolist():\n",
    "    cr_journal, cr_journal_short,cr_issn, publisher, pubdate = '','','','',''\n",
    "    \n",
    "    if doi in cr_dict:\n",
    "        item = cr_dict[doi]\n",
    "    \n",
    "        if 'ISSN' in item:\n",
    "            cr_issn = '; '.join(item['ISSN'])\n",
    "    \n",
    "        pubdate = str(cr_earliest_date(item))[:10]\n",
    "        \n",
    "        if 'publisher' in item:\n",
    "            publisher = item['publisher']\n",
    "            \n",
    "        if 'container-title' in item:\n",
    "            cr_journal = item['container-title'][0]\n",
    "            \n",
    "        if 'short-container-title' in item:\n",
    "            cr_journal_short = item['short-container-title'][0]\n",
    "        \n",
    "    else:\n",
    "        pass\n",
    "    \n",
    "    cr_journals.append(cr_journal)\n",
    "    cr_journals_short.append(cr_journal_short)\n",
    "    cr_issns.append(cr_issn)\n",
    "    publishers.append(publisher)\n",
    "    pubdates.append(pubdate)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13202, 13)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['journal'] = cr_journals\n",
    "df['journal-short'] = cr_journals_short\n",
    "df['pubdate'] = pubdates\n",
    "df['issns'] = cr_issns\n",
    "df['publisher'] = publishers\n",
    "\n",
    "df.shape\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pid</th>\n",
       "      <th>doi</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>authors</th>\n",
       "      <th>venue</th>\n",
       "      <th>year</th>\n",
       "      <th>tiabs</th>\n",
       "      <th>journal</th>\n",
       "      <th>journal-short</th>\n",
       "      <th>pubdate</th>\n",
       "      <th>issns</th>\n",
       "      <th>publisher</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>6495</td>\n",
       "      <td>07899a6af647ba443bf04599aa35d111d0d08cb3</td>\n",
       "      <td>10.3390/brainsci7070078</td>\n",
       "      <td>Multiple Sclerosis: Immunopathology and Treatm...</td>\n",
       "      <td>The treatment of multiple sclerosis (MS) has c...</td>\n",
       "      <td>Dargahi, Narges; Katsara, Maria; Tselios, Theo...</td>\n",
       "      <td>Brain sciences</td>\n",
       "      <td>2017</td>\n",
       "      <td>Multiple Sclerosis: Immunopathology and Treatm...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11297</td>\n",
       "      <td>4ee0f8eb93dda0ff3512556fbb784c3859cc156f</td>\n",
       "      <td></td>\n",
       "      <td>Jet set pets: examining the zoonosis risk in a...</td>\n",
       "      <td>Ownership of companion animals or pets is popu...</td>\n",
       "      <td>Fooks, Anthony; Johnson, Nicholas</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Jet set pets: examining the zoonosis risk in a...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10458</td>\n",
       "      <td>963c0263d2acd998d323aabd13c7c103bb44f6d3</td>\n",
       "      <td>10.1038/ncomms5323</td>\n",
       "      <td>Reconstructing propagation networks with natur...</td>\n",
       "      <td>Our ability to uncover complex network structu...</td>\n",
       "      <td>Shen, Zhesi; Wang, Wen-Xu; Fan, Ying; Di, Zeng...</td>\n",
       "      <td>Nature communications</td>\n",
       "      <td>2014</td>\n",
       "      <td>Reconstructing propagation networks with natur...</td>\n",
       "      <td>Nature Communications</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>2014-07-11</td>\n",
       "      <td>2041-1723</td>\n",
       "      <td>Springer Science and Business Media LLC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2801</td>\n",
       "      <td>5e11e3105842cf7cedf10e26bff170a52c23cf00</td>\n",
       "      <td>10.1371/journal.ppat.1003124</td>\n",
       "      <td>IFITM Proteins Restrict Viral Membrane Hemifusion</td>\n",
       "      <td>The interferon-inducible transmembrane (IFITM)...</td>\n",
       "      <td>Li, K; Markosyan, R; Zheng, Y-M; Golfetto, O; ...</td>\n",
       "      <td>PLoS pathogens</td>\n",
       "      <td>2013</td>\n",
       "      <td>IFITM Proteins Restrict Viral Membrane Hemifus...</td>\n",
       "      <td>PLoS Pathogens</td>\n",
       "      <td>PLoS Pathog</td>\n",
       "      <td>2013-01-24</td>\n",
       "      <td>1553-7374</td>\n",
       "      <td>Public Library of Science (PLoS)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>290</td>\n",
       "      <td>cc41c51293ed24a57ddf467e462ec9399463ee33</td>\n",
       "      <td></td>\n",
       "      <td>No more business as usual: agile and effective...</td>\n",
       "      <td>5,6 , usegalaxy.be 7,8 and hyphy.org 9 develop...</td>\n",
       "      <td>Org, Usegalaxy; Baker, Dannon; Van Den Beek, M...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No more business as usual: agile and effective...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9245</td>\n",
       "      <td>89fc23ed55ad474b4b4056c0a3dadd315573f57e</td>\n",
       "      <td></td>\n",
       "      <td>A Novel Bacterium-Like Particle Vaccine Displa...</td>\n",
       "      <td>Middle East respiratory syndrome coronavirus (...</td>\n",
       "      <td>Li, Entao; Chi, Hang; Huang, Pei; Yan, Feihu; ...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>A Novel Bacterium-Like Particle Vaccine Displa...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12218</td>\n",
       "      <td>d66d18ad33170f29a2279d8cc4ca3e31bdee67f0</td>\n",
       "      <td>10.3201/eid1201.051024</td>\n",
       "      <td>H5N1 Outbreaks and Enzootic Influenza</td>\n",
       "      <td>Ongoing outbreaks of H5N1 avian influenza in m...</td>\n",
       "      <td>Webster, Robert; Peiris, Malik; Chen, Honglin;...</td>\n",
       "      <td>Emerging infectious diseases</td>\n",
       "      <td>2006</td>\n",
       "      <td>H5N1 Outbreaks and Enzootic Influenza. Ongoing...</td>\n",
       "      <td>Emerging Infectious Diseases</td>\n",
       "      <td>Emerg. Infect. Dis.</td>\n",
       "      <td>2012-02-03</td>\n",
       "      <td>1080-6040; 1080-6059</td>\n",
       "      <td>Centers for Disease Control and Prevention (CDC)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12376</td>\n",
       "      <td>76afb423879efd78bcd26ac2be9086727030f990</td>\n",
       "      <td></td>\n",
       "      <td>Four cases of severe acute respiratory syndrom...</td>\n",
       "      <td>in the city of Guangzhou, Guangdong Province, ...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>Four cases of severe acute respiratory syndrom...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10181</td>\n",
       "      <td>b7d7db5b12584c5fab0c2244a27d0653d8a59f7d</td>\n",
       "      <td>10.3389/fmicb.2012.00068</td>\n",
       "      <td>Mouse Hepatitis Virus Receptor as a Determinan...</td>\n",
       "      <td>In this review, we report that the receptor of...</td>\n",
       "      <td>Taguchi, Fumihiro; Hirai-Yuki, Asuka</td>\n",
       "      <td>Front. Microbio.</td>\n",
       "      <td>2012</td>\n",
       "      <td>Mouse Hepatitis Virus Receptor as a Determinan...</td>\n",
       "      <td>Frontiers in Microbiology</td>\n",
       "      <td>Front. Microbio.</td>\n",
       "      <td>2012-02-24</td>\n",
       "      <td>1664-302X</td>\n",
       "      <td>Frontiers Media SA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1194</td>\n",
       "      <td>4cd8e92fcab8da2c840ff39f14ca35bf5ba6e444</td>\n",
       "      <td>10.1186/1471-2288-8-20</td>\n",
       "      <td>Graphical presentation of diagnostic information</td>\n",
       "      <td>BackgroundGraphical displays of results allow ...</td>\n",
       "      <td>Whiting, Penny; Ac, Jonathan; , Sterne; Westwo...</td>\n",
       "      <td>BMC medical research methodology</td>\n",
       "      <td>2008</td>\n",
       "      <td>Graphical presentation of diagnostic informati...</td>\n",
       "      <td>BMC Medical Research Methodology</td>\n",
       "      <td>BMC Med Res Methodol</td>\n",
       "      <td>2008-04-11</td>\n",
       "      <td>1471-2288</td>\n",
       "      <td>Springer Science and Business Media LLC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            pid                           doi  \\\n",
       "6495   07899a6af647ba443bf04599aa35d111d0d08cb3       10.3390/brainsci7070078   \n",
       "11297  4ee0f8eb93dda0ff3512556fbb784c3859cc156f                                 \n",
       "10458  963c0263d2acd998d323aabd13c7c103bb44f6d3            10.1038/ncomms5323   \n",
       "2801   5e11e3105842cf7cedf10e26bff170a52c23cf00  10.1371/journal.ppat.1003124   \n",
       "290    cc41c51293ed24a57ddf467e462ec9399463ee33                                 \n",
       "9245   89fc23ed55ad474b4b4056c0a3dadd315573f57e                                 \n",
       "12218  d66d18ad33170f29a2279d8cc4ca3e31bdee67f0        10.3201/eid1201.051024   \n",
       "12376  76afb423879efd78bcd26ac2be9086727030f990                                 \n",
       "10181  b7d7db5b12584c5fab0c2244a27d0653d8a59f7d      10.3389/fmicb.2012.00068   \n",
       "1194   4cd8e92fcab8da2c840ff39f14ca35bf5ba6e444        10.1186/1471-2288-8-20   \n",
       "\n",
       "                                                   title  \\\n",
       "6495   Multiple Sclerosis: Immunopathology and Treatm...   \n",
       "11297  Jet set pets: examining the zoonosis risk in a...   \n",
       "10458  Reconstructing propagation networks with natur...   \n",
       "2801   IFITM Proteins Restrict Viral Membrane Hemifusion   \n",
       "290    No more business as usual: agile and effective...   \n",
       "9245   A Novel Bacterium-Like Particle Vaccine Displa...   \n",
       "12218              H5N1 Outbreaks and Enzootic Influenza   \n",
       "12376  Four cases of severe acute respiratory syndrom...   \n",
       "10181  Mouse Hepatitis Virus Receptor as a Determinan...   \n",
       "1194    Graphical presentation of diagnostic information   \n",
       "\n",
       "                                                abstract  \\\n",
       "6495   The treatment of multiple sclerosis (MS) has c...   \n",
       "11297  Ownership of companion animals or pets is popu...   \n",
       "10458  Our ability to uncover complex network structu...   \n",
       "2801   The interferon-inducible transmembrane (IFITM)...   \n",
       "290    5,6 , usegalaxy.be 7,8 and hyphy.org 9 develop...   \n",
       "9245   Middle East respiratory syndrome coronavirus (...   \n",
       "12218  Ongoing outbreaks of H5N1 avian influenza in m...   \n",
       "12376  in the city of Guangzhou, Guangdong Province, ...   \n",
       "10181  In this review, we report that the receptor of...   \n",
       "1194   BackgroundGraphical displays of results allow ...   \n",
       "\n",
       "                                                 authors  \\\n",
       "6495   Dargahi, Narges; Katsara, Maria; Tselios, Theo...   \n",
       "11297                  Fooks, Anthony; Johnson, Nicholas   \n",
       "10458  Shen, Zhesi; Wang, Wen-Xu; Fan, Ying; Di, Zeng...   \n",
       "2801   Li, K; Markosyan, R; Zheng, Y-M; Golfetto, O; ...   \n",
       "290    Org, Usegalaxy; Baker, Dannon; Van Den Beek, M...   \n",
       "9245   Li, Entao; Chi, Hang; Huang, Pei; Yan, Feihu; ...   \n",
       "12218  Webster, Robert; Peiris, Malik; Chen, Honglin;...   \n",
       "12376                                                      \n",
       "10181               Taguchi, Fumihiro; Hirai-Yuki, Asuka   \n",
       "1194   Whiting, Penny; Ac, Jonathan; , Sterne; Westwo...   \n",
       "\n",
       "                                  venue  year  \\\n",
       "6495                     Brain sciences  2017   \n",
       "11297                                           \n",
       "10458             Nature communications  2014   \n",
       "2801                     PLoS pathogens  2013   \n",
       "290                                             \n",
       "9245                                            \n",
       "12218      Emerging infectious diseases  2006   \n",
       "12376                                           \n",
       "10181                  Front. Microbio.  2012   \n",
       "1194   BMC medical research methodology  2008   \n",
       "\n",
       "                                                   tiabs  \\\n",
       "6495   Multiple Sclerosis: Immunopathology and Treatm...   \n",
       "11297  Jet set pets: examining the zoonosis risk in a...   \n",
       "10458  Reconstructing propagation networks with natur...   \n",
       "2801   IFITM Proteins Restrict Viral Membrane Hemifus...   \n",
       "290    No more business as usual: agile and effective...   \n",
       "9245   A Novel Bacterium-Like Particle Vaccine Displa...   \n",
       "12218  H5N1 Outbreaks and Enzootic Influenza. Ongoing...   \n",
       "12376  Four cases of severe acute respiratory syndrom...   \n",
       "10181  Mouse Hepatitis Virus Receptor as a Determinan...   \n",
       "1194   Graphical presentation of diagnostic informati...   \n",
       "\n",
       "                                journal         journal-short     pubdate  \\\n",
       "6495                                                                        \n",
       "11297                                                                       \n",
       "10458             Nature Communications            Nat Commun  2014-07-11   \n",
       "2801                     PLoS Pathogens           PLoS Pathog  2013-01-24   \n",
       "290                                                                         \n",
       "9245                                                                        \n",
       "12218      Emerging Infectious Diseases   Emerg. Infect. Dis.  2012-02-03   \n",
       "12376                                                                       \n",
       "10181         Frontiers in Microbiology      Front. Microbio.  2012-02-24   \n",
       "1194   BMC Medical Research Methodology  BMC Med Res Methodol  2008-04-11   \n",
       "\n",
       "                      issns                                         publisher  \n",
       "6495                                                                           \n",
       "11297                                                                          \n",
       "10458             2041-1723           Springer Science and Business Media LLC  \n",
       "2801              1553-7374                  Public Library of Science (PLoS)  \n",
       "290                                                                            \n",
       "9245                                                                           \n",
       "12218  1080-6040; 1080-6059  Centers for Disease Control and Prevention (CDC)  \n",
       "12376                                                                          \n",
       "10181             1664-302X                                Frontiers Media SA  \n",
       "1194              1471-2288           Springer Science and Business Media LLC  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df[[('SAGE' in x) for x in df['publisher']]].to_excel('output/SAGE_in_s2.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-f0e784d0234c>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'publisher'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvalue_counts\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "df['publisher'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
